---
title: "GWAS SumStats Software"
# author:
  # - name: Jacob Knyspel
  #   url: https://jacobknyspel.github.io/
  #   orcid: 0000-0001-5600-4826
  #   affiliation: Social, Genetic & Developmental Psychiatry Centre, Kings College London
date: 01-13-2025
categories: [software, genetics]
# citation: 
#   url: https://jacobknyspel.github.io/posts/YYYY-MM-DD-test-post/ 
draft: false # setting this to `true` will prevent your post from appearing on your listing page until you're ready!
tbl-colwidths: [0,0,14,0,0]
image: genome.jpg
---

## Background

Researchers use summary statistics produced by Genome-Wide Association Studies (GWAS) for a broad range of downstream analyses. In recent years, available software for handling and using GWAS summary statistics has proliferated considerably, making it increasingly difficult to keep track of this software and to make decisions about which software to use. Panagiota Kontou and Pantelis Bagos recently conducted a systematic review of GWAS summary statistic software published or in pre-print as of December 2023 ([see here](https://doi.org/10.1186/s13040-024-00385-x)). Here I provide an overview of the software found in this systematic review, with brief descriptions of each software taken from the supplementary materials.

::: callout-tip
## The systematic review did not include software for calculating polygenic scores (e.g. LDpred2, PRS-CS) and so I might add this in the future.
:::

::: callout-note
## If there are any errors or important omissions, let me know.
:::

## Data management

### Quality control

+-----------------+-------+-------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| Name            | Year  | Links                                                                                                                         | Description                                                                                                                                                                                                                                        | Language |
+=================+=======+===============================================================================================================================+====================================================================================================================================================================================================================================================+==========+
| GWAS-SSF        | 2023  | [Paper](https://doi.org/10.1101/2022.07.15.500230)                                                                            | Specifications for the first version of the GWAS-SSF format, which was developed to meet the requirements discussed with the community. GWAS-SSF consists of a tab-separated data file with well-defined fields and an accompanying metadata file. | Python   |
|                 |       |                                                                                                                               |                                                                                                                                                                                                                                                    |          |
|                 |       | [GitHub](https://github.com/EBISPOT/gwas-summary-statistics-standard)                                                         |                                                                                                                                                                                                                                                    |          |
+-----------------+-------+-------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GWASlab         | 2023  | [Paper](https://doi.org/10.51094/jxiv.370)                                                                                    | A toolkit for handling GWAS summary statistics. Offers functionalities for converting most formats, standardization, normalization, harmonization, filtering and visualization.                                                                    | Python   |
|                 |       |                                                                                                                               |                                                                                                                                                                                                                                                    |          |
|                 |       | [GitHub](https://cloufield.github.io/gwaslab/)                                                                                |                                                                                                                                                                                                                                                    |          |
+-----------------+-------+-------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GQS             | 2023  | [Paper](https://doi.org/10.1093/bioinformatics/btad004)                                                                       | Identifies suspicious regions and prevents erroneous interpretations. Assesses all measured SNPs in LD and compares the significance of trait association of each SNP to its LD value.                                                             | Python   |
|                 |       |                                                                                                                               |                                                                                                                                                                                                                                                    |          |
|                 |       | [GitHub](https://github.com/Xswapnil/GQS/)                                                                                    |                                                                                                                                                                                                                                                    |          |
+-----------------+-------+-------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| EXTminus23andMe | 2023  | [Paper](https://doi.org/10.1007/s10519-023-10152-z)                                                                           | A tool to evaluate the quality of summary statistics after data removal and the suitability of these downsampled summary statistics for typical follow-up genetic analyses.                                                                        | R        |
|                 |       |                                                                                                                               |                                                                                                                                                                                                                                                    |          |
|                 |       | [GitHub](https://github.com/Camzcamz/EXTminus23andMe)                                                                         |                                                                                                                                                                                                                                                    |          |
+-----------------+-------+-------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| SumStatsRehab   | 2022  | [Paper](https://doi.org/10.1186/s12859-022-04920-7)                                                                           | A tool for data validation, restoration of missing data, correction and formatting.                                                                                                                                                                | Python   |
|                 |       |                                                                                                                               |                                                                                                                                                                                                                                                    |          |
|                 |       | [GitHub](https://github.com/Kukuster/SumStatsRehab)                                                                           |                                                                                                                                                                                                                                                    |          |
+-----------------+-------+-------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MungeSumstats   | 2021  | [Paper](https://doi.org/10.1093/bioinformatics/btab665)                                                                       | A tool for the standardization and quality control of GWAS summary statistics. It can handle the most common summary statistic formats.                                                                                                            | R        |
|                 |       |                                                                                                                               |                                                                                                                                                                                                                                                    |          |
|                 |       | [GitHub](https://github.com/neurogenomics/MungeSumstats)                                                                      |                                                                                                                                                                                                                                                    |          |
+-----------------+-------+-------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GWASinspector   | 2021  | [Paper](https://doi.org/10.1093/bioinformatics/btaa1084)                                                                      | Developed to facilitate and streamline this process and provide the user with a comprehensive report. It will also generate cleaned, harmonized GWAS files ready for meta-analysis.                                                                | R        |
|                 |       |                                                                                                                               |                                                                                                                                                                                                                                                    |          |
|                 |       | [Webpage](https://gwasinspector.com/)                                                                                         |                                                                                                                                                                                                                                                    |          |
+-----------------+-------+-------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| VCF             | 2021  | [Paper](https://doi.org/10.1186/s13059-020-02248-0)                                                                           | The variant call format is used to store GWAS summary statistics along with open-source tools to be uses in downstream analyses.                                                                                                                   | Python   |
|                 |       |                                                                                                                               |                                                                                                                                                                                                                                                    |          |
|                 |       | [GitHub](https://github.com/MRCIEU/gwas-vcf-specification)                                                                    |                                                                                                                                                                                                                                                    |          |
+-----------------+-------+-------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| DENTIST         | 2021  | [Paper](https://doi.org/10.1038/s41467-021-27438-7)                                                                           | Leverages LD among SNPs to detect and eliminate errors in GWAS or LD reference and heterogeneity between the two.                                                                                                                                  | C/C++    |
|                 |       |                                                                                                                               |                                                                                                                                                                                                                                                    |          |
|                 |       | [GitHub](https://github.com/Yves-CHEN/DENTIST)                                                                                |                                                                                                                                                                                                                                                    |          |
+-----------------+-------+-------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GEAR            | 2016  | [Paper](https://doi.org/10.1038/ejhg.2016.106)                                                                                | A tool that contains functions to identify significant sample overlap or heterogeneity between pairs of cohorts.                                                                                                                                   | Java     |
|                 |       |                                                                                                                               |                                                                                                                                                                                                                                                    |          |
|                 |       | [GitHub](https://github.com/syntheke/GEAR)                                                                                    |                                                                                                                                                                                                                                                    |          |
+-----------------+-------+-------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| PLINK (1.9)     | 2015  | [Paper](https://doi.org/10.1186/s13742-015-0047-8)                                                                            | A versatile program which supports data management, quality control, and common statistical computations.                                                                                                                                          | C/C++    |
|                 |       |                                                                                                                               |                                                                                                                                                                                                                                                    |          |
|                 |       | [Webpage](https://www.cog-genomics.org/plink/1.9/)                                                                            |                                                                                                                                                                                                                                                    |          |
+-----------------+-------+-------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| EasyQC          | 2014  | [Paper](https://doi.org/10.1038/nprot.2014.071)                                                                               | A general protocol for conducting meta-analysis and carrying out QC to minimize errors and to guarantee maximum use of data.                                                                                                                       | R        |
|                 |       |                                                                                                                               |                                                                                                                                                                                                                                                    |          |
|                 |       | [Webpage](https://www.uni-regensburg.de/medizin/epidemiologie-praeventivmedizin/genetische-epidemiologie/software/index.html) |                                                                                                                                                                                                                                                    |          |
+-----------------+-------+-------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| QCGWAS          | 2014  | [Paper](10.1093/bioinformatics/btt745)                                                                                        | Automates the quality control of GWAS result files. Its main purpose is to facilitate the quality control of a large number of such files before meta-analysis.                                                                                    | R        |
|                 |       |                                                                                                                               |                                                                                                                                                                                                                                                    |          |
|                 |       | [CRAN](https://cran.r-project.org/web/packages/QCGWAS/index.html)                                                             |                                                                                                                                                                                                                                                    |          |
+-----------------+-------+-------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GWAtoolbox      | 2012  | [Paper](https://doi.org/10.1093/bioinformatics/btr679)                                                                        | Contains three particular data quality aspects: data formatting, quality of the GWAS results and data consistency across studies. It was removed from CRAN in 2014 due to attribution issues.                                                      | R        |
|                 |       |                                                                                                                               |                                                                                                                                                                                                                                                    |          |
|                 |       | [GitHub](https://github.com/cran/GWAtoolbox)                                                                                  |                                                                                                                                                                                                                                                    |          |
+-----------------+-------+-------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+

: Software are sorted by most to least recent {.striped}

### Reconstruction

+--------------+-------+---------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| Name         | Year  | Links                                                         | Description                                                                                                                                                    | Language |
+==============+=======+===============================================================+================================================================================================================================================================+==========+
| spkmt        | 2023  | [Paper](https://doi.org/10.1186/s13104-023-06438-4)           | Method to derive GWAS summary statistics for one parent when observations have only been made on the offspring and another parent.                             | R        |
|              |       |                                                               |                                                                                                                                                                |          |
|              |       | [OSF](https://osf.io/spkmt/)                                  |                                                                                                                                                                |          |
+--------------+-------+---------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| ReACt        | 2022  | [Paper](https://doi.org/10.1038/s41598-022-12185-6)           | Performs genotype reconstruction for case-control GWAS summary statistics. It includes three modules: meta-analysis, group PRS and case-case GWAS.             | C/C++    |
|              |       |                                                               |                                                                                                                                                                |          |
|              |       | [GitHub](https://github.com/Paschou-Lab/ReACt)                |                                                                                                                                                                |          |
+--------------+-------+---------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| Metasubtract | 2020  | [Paper](https://doi.org/10.1093/bioinformatics/btaa570)       | Subtracts the results of a validation cohort from meta-GWAS summary statistics analytically.                                                                   | R        |
|              |       |                                                               |                                                                                                                                                                |          |
|              |       | [CRAN](https://cran.r-project.org/web/packages/MetaSubtract/) |                                                                                                                                                                |          |
+--------------+-------+---------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| simGWAS      | 2019  | [Paper](https://doi.org/10.1093/bioinformatics/bty898)        | Simulates GWAS summary data without individual data as an intermediate step.                                                                                   | R        |
|              |       |                                                               |                                                                                                                                                                |          |
|              |       | [GitHub](https://chr1swallace.github.io/simGWAS/index.html)   |                                                                                                                                                                |          |
+--------------+-------+---------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| LMOR         | 2018  | [Paper](https://doi.org/10.1534/genetics.117.300360)          | Performs transformations from genetic effects estimated under a linear mixed model to odds ratios that only rely on summary statistics.                        | R        |
|              |       |                                                               |                                                                                                                                                                |          |
|              |       | [GitHub](https://github.com/lukelloydjones/ORShiny)           |                                                                                                                                                                |          |
+--------------+-------+---------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| OATH         | 2017  | [Paper](https://doi.org/10.1534/g3.116.038877)                | Reproduces reported results from a GWAS and recovers under-reported results from other alternative models with a different combination of nuisance parameters. | Java     |
|              |       |                                                               |                                                                                                                                                                |          |
|              |       | [GitHub](https://github.com/gc5k/GEAR)                        |                                                                                                                                                                |          |
+--------------+-------+---------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+

: Software are sorted by most to least recent {.striped}

### Imputation

+--------------+-------+----------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| Name         | Year  | Links                                                          | Description                                                                                                                                                                           | Language |
+==============+=======+================================================================+=======================================================================================================================================================================================+==========+
| LS-META      | 2023  | [Paper](https://doi.org/10.1093/hmg/ddad097)                   | Imputes both genetic and environmental components of a trait using both SNP-trait and omics-trait association summary data.                                                           | R        |
|              |       |                                                                |                                                                                                                                                                                       |          |
|              |       | [GitHub](https://github.com/ren328/LS-Meta)                    |                                                                                                                                                                                       |          |
+--------------+-------+----------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| LSimputing   | 2023  | [Paper](https://doi.org/10.1016/j.xhgg.2023.100197)            | A nonparametric method for large-scale imputation of the genotype effects. If a sample of IPD is available the method allows for nonlinear SNP-trait associations and predictions.    | R        |
|              |       |                                                                |                                                                                                                                                                                       |          |
|              |       | [GitHub](https://github.com/ren328/LSimputing)                 |                                                                                                                                                                                       |          |
+--------------+-------+----------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| RAISS        | 2019  | [Paper](https://doi.org/10.1093/bioinformatics/btz466)         | Uses LD and the multivariate normal distriburion along with several optimizations.                                                                                                    | Python   |
|              |       |                                                                |                                                                                                                                                                                       |          |
|              |       | [GitLab](https://gitlab.pasteur.fr/statistical-genetics/raiss) |                                                                                                                                                                                       |          |
+--------------+-------+----------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| SSimp        | 2018  | [Paper](https://doi.org/10.1371/journal.pgen.1007371)          | Uses the multivariate normal distribution and LD from an external source.                                                                                                             | C/C++    |
|              |       |                                                                |                                                                                                                                                                                       |          |
|              |       | [GitHub](https://github.com/zkutalik/ssimp_software)           |                                                                                                                                                                                       |          |
+--------------+-------+----------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| ARDISS       | 2018  | [Paper](https://doi.org/10.1093/bioinformatics/bty596)         | Imputes missing summary statistics in mixed-ethnicity cohorts through Gaussian process regression and automatic relevance determination.                                              | Python   |
|              |       |                                                                |                                                                                                                                                                                       |          |
|              |       | [GitHub](https://github.com/BorgwardtLab/ARDISS)               |                                                                                                                                                                                       |          |
+--------------+-------+----------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| FAPI         | 2016  | [Paper](https://doi.org/10.1038/ejhg.2015.190)                 | Fast and accurate p-value imputation method that utilizes summary statistics of common variants. Its computational cost is linear with the number of untyped variants.                | Exe      |
|              |       |                                                                |                                                                                                                                                                                       |          |
|              |       | [Webpage](https://pmglab.top/fapi/)                            |                                                                                                                                                                                       |          |
+--------------+-------+----------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| DIST/DISTMIX | 2015  | [Paper](https://doi.org/10.1093/bioinformatics/btv348)         | Uses the multivariate normal distribution and the correlation structure from a relevant reference population. DISTMIX is the extension to mixed ethnicity cohorts.                    | Exe      |
|              |       |                                                                |                                                                                                                                                                                       |          |
|              |       | [GitHub](https://dleelab.github.io/distmix/)                   |                                                                                                                                                                                       |          |
+--------------+-------+----------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| DISSCO       | 2015  | [Paper](https://doi.org/10.1093/bioinformatics/btv168)         | Uses the multivariate normal distribution and LD, and allows for covariates.                                                                                                          | Java     |
|              |       |                                                                |                                                                                                                                                                                       |          |
|              |       | [Webpage](https://yunliweb.its.unc.edu/DISSCO/)                |                                                                                                                                                                                       |          |
+--------------+-------+----------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| Adapt-Mix    | 2015  | [Paper](https://doi.org/10.1093/bioinformatics/btv230)         | Combines information across all available reference panels to produce estimates of local genetic correlation structure for summary statistics-based methods in arbitrary populations. | Python   |
|              |       |                                                                |                                                                                                                                                                                       |          |
|              |       | [GitHub](https://github.com/dpark27/adapt_mix)                 |                                                                                                                                                                                       |          |
+--------------+-------+----------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| impG         | 2014  | [Paper](https://doi.org/10.1093/bioinformatics/btu416)         | Uses the multivariate normal distribution and LD from an external source.                                                                                                             | C/C++    |
|              |       |                                                                |                                                                                                                                                                                       |          |
|              |       | [Webpage](https://bogdan.dgsom.ucla.edu/pages/impg/)           |                                                                                                                                                                                       |          |
+--------------+-------+----------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+

: Software are sorted by most to least recent {.striped}

## Single trait analysis

### Meta-analysis

+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| Name            | Year  | Links                                                                            | Description                                                                                                                                                                                                              | Language  |
+=================+=======+==================================================================================+==========================================================================================================================================================================================================================+===========+
| MAJAR           | 2023  | [Paper](https://doi.org/10.1177/09622802231188519)                               | Method to jointly test prognostic and predictive effects in meta-analysis without the need of using an independent cohort for replication of the detected biomarkers.                                                    | R         |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [GitHub](https://github.com/JustinaXie/MAJAR)                                    |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| sPLINK          | 2022  | [Paper](https://doi.org/10.1186/s13059-021-02562-1)                              | Performs privacy-aware GWAS on distributed datasets while preserving accuracy.                                                                                                                                           | Python    |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [Webpage](https://exbio.wzw.tum.de/splink/)                                      |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| GWASmeta        | 2022  | [Paper](https://doi.org/10.1371/journal.pcbi.1009948)                            | A method for the optimal ABF in the GWAS meta-analysis, SMetABF. Uses shotgun stochastic search to improve the Bayesian GWAS meta-analysis framework.                                                                    | R         |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [GitHub](https://github.com/sjl-sjtu/GWAS_meta)                                  |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| GCPBayes        | 2021  | [Paper](https://doi.org/10.1002/sim.8855)                                        | Bayesian meta-analysis methods for pleiotropy that extend CPBayes to the gene or pathway level.                                                                                                                          | R         |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [GitHub](https://github.com/tbaghfalaki/GCPBayes)                                |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| nGWAMA          | 2019  | [Paper](https://doi.org/10.1038/s41588-018-0320-8)                               | Performs multivariate meta-analysis correcting for sample overlap.                                                                                                                                                       | R         |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [GitHub](https://github.com/baselmans/multivariate_GWAMA)                        |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| MetABF          | 2019  | [Paper](https://doi.org/10.1002/gepi.22202)                                      | A simple Bayesian framework for performing integrative meta-analysis across multiple GWAS.                                                                                                                               | R         |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [GitHub](https://github.com/trochet/metabf)                                      |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| CPBayes         | 2018  | [Paper](https://doi.org/10.1371/journal.pgen.1007139)                            | Bayesian method for studying cross-phenotype genetic associations.                                                                                                                                                       | R         |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [GitHub](https://github.com/ArunabhaCodes/CPBayes)                               |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| metaUSAT        | 2018  | [Paper](https://doi.org/10.1002/gepi.22105)                                      | A method for multiple traits. It is robust to the association structure of correlated traits. It can also be used to analyze a single trait over multiple studies, accounting for overlapping samples.                   | R         |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [GitHub](https://github.com/RayDebashree/metaUSAT)                               |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| rareMETALS      | 2018  | [Paper](https://doi.org/10.1371/journal.pgen.1007452)                            | Works even when the data contain large amounts of missing values. Uses a score statistic called PCBS (partial correlation based score statistic) for conditional analysis of single-variant and gene-level associations. | R         |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [Webpage](https://genome.sph.umich.edu/wiki/RareMETALS)                          |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| meta-simulation | 2018  | [Paper](https://doi.org/10.1016/j.mgene.2018.02.004)                             | A tool to implement an alternate strategy for the additive model based on simulating data for the individual studies.                                                                                                    | R         |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| rfdr            | 2018  | [Paper](https://doi.org/10.1177/0962280216687168)                                | A method for replication of GWAS. Provides the most powerful significance levels when controlling the FDR in the two-stage study.                                                                                        | R         |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [Webpage](https://bioinformatics.hkust.edu.hk/RFdr.html)                         |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| GWAR            | 2017  | [Paper](https://doi.org/10.1093/bioinformatics/btx008)                           | Analysis and meta-analysis of GWAS using standard as well as robust methods (MAX, MIN2, MERT).                                                                                                                           | Stata/web |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [Webpage](https://www.compgen.org/tools/GWAR)                                    |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| metaGAP         | 2017  | [Paper](https://doi.org/10.1371/journal.pgen.1006495)                            | A versatile tool for calculating the statistical power of a meta-analysis of GWAS results and of the polygenic-score R² in a hold-out sample.                                                                            | Web       |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [Webpage](https://www.devlaming.eu/metagap.html)                                 |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| jlfdr           | 2017  | [Paper](https://doi.org/10.1093/bioinformatics/btw690)                           | A joint analysis method based on controlling the joint local false discovery rate.                                                                                                                                       | R         |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [Webpage](https://bioinformatics.hkust.edu.hk/Jlfdr.html)                        |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| RRate           | 2017  | [Paper](https://doi.org/10.1093/bib/bbw091)                                      | A Bayesian probabilistic measure of the replication rate with which we can determine the sample size of the replication study and to check the consistency between the primary and the replication study.                | R         |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [Webpage](https://bioinformatics.hkust.edu.hk/RRate.html)                        |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| metaCCA         | 2016  | [Paper](https://doi.org/10.1093/bioinformatics/btw052)                           | Multivariate analysis and meta-anaysis of GWAS. It uses canonical correlation analysis and employs a covariance shrinkage algorithm to achieve robustness.                                                               | R         |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [Bioconductor](https://bioconductor.org/packages/release/bioc/html/metaCCA.html) |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| PLINK (1.9)     | 2015  | [Paper](https://doi.org/10.1186/s13742-015-0047-8)                               | A versatile program which supports data management, quality control, and common statistical computations including meta-analysis.                                                                                        | C/C++     |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [Webpage](https://www.cog-genomics.org/plink/1.9/)                               |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| XPEB            | 2015  | [Paper](https://doi.org/10.1016/j.ajhg.2015.03.008)                              | An empirical Bayes approach to improve the power of GWAS in a minority population by exploiting information from another ethnic population.                                                                              | R         |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [Webpage](https://med.stanford.edu/tanglab/software/XPEB.html)                   |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| CPASSOC         | 2015  | [Paper](https://doi.org/10.1016/j.ajhg.2014.11.011)                              | Method applicable to a multivariate phenotype containing any type of components including continuous, categorical and survival phenotypes, as well as to samples consisting of families or unrelated samples.            | R         |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [Webpage](https://hal.case.edu/~xxz10/zhu-web/)                                  |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| metaSKAT        | 2013  | [Paper](https://doi.org/10.1016/j.ajhg.2013.05.010)                              | Extensions of the Burden Test, SKAT and Optimal SKAT (SKAT-O) for multiple studies.                                                                                                                                      | R         |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [GitHub](https://github.com/lin-lab/MetaSKAT)                                    |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| METACARPA       | 2013  | [Paper](https://doi.org/10.1142/9789814447973_0023)                              | Meta-analysis of GWAS with overlapping or related samples, when details of the overlap or relatedness are unknown.                                                                                                       | C/C++     |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [GitHub](https://github.com/hmgu-itg/metacarpa)                                  |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| YAMAS           | 2012  | [Paper](https://doi.org/10.1186/1471-2105-13-231)                                | Meta-analysis including missing SNPs identified with LD (proxy SNPs).                                                                                                                                                    | C/C++     |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [GitHub](https://github.com/cmeesters/yamas)                                     |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| METAL           | 2010  | [Paper](https://doi.org/10.1093/bioinformatics/btq340)                           | A versatile and efficient tool for meta-analysis of GWAS. It can combine test statistics and standard errors, or p-values across studies.                                                                                | C/C++     |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [Webpage](https://csg.sph.umich.edu/abecasis/Metal/)                             |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| MAGENTA         | 2010  | [Paper](https://doi.org/10.1371/journal.pgen.1001058)                            | Meta-analysis with gene set enrichment analysis (GSEA).                                                                                                                                                                  | Matlab    |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [Webpage](https://software.broadinstitute.org/mpg/magenta/)                      |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| GWAMA           | 2010  | [Paper](https://doi.org/10.1186/1471-2105-11-288)                                | A flexible, open-source tool for meta-analysis of GWAS. It incorporates a variety of error trapping facilities, and provides a range of meta-analysis summary statistics.                                                | Exe       |
|                 |       |                                                                                  |                                                                                                                                                                                                                          |           |
|                 |       | [Webpage](https://genomics.ut.ee/en/tools)                                       |                                                                                                                                                                                                                          |           |
+-----------------+-------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+

: Software are sorted by most to least recent {.striped}

### Heritability

+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| Name                      | Year  | Links                                                                 | Description                                                                                                                                                                                                                                          | Language |
+===========================+=======+=======================================================================+======================================================================================================================================================================================================================================================+==========+
| HEELS                     | 2023  | [Paper](https://doi.org/10.1038/s41467-023-43565-9)                   | Uses REML to produce accurate and precise local heritability estimates.                                                                                                                                                                              | Python   |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitHub](https://github.com/huilisabrina/HEELS)                       |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| HAMSTA                    | 2023  | [Paper](https://doi.org/10.1016/j.ajhg.2023.09.012)                   | Estimates the heritability explained by local ancestry in admixture mapping studies. It also quantifies inflation in test statistics that is not contributed by local ancestry effects, and determines significance threshold for admixture mapping. | Python   |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitHub](https://tszfungc.github.io/HAMSTA/)                          |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| LDER                      | 2022  | [Paper](https://doi.org/10.1016/j.ajhg.2022.03.013)                   | Extends the LDSC method making full use of the information from the LD matrix and provides more accurate estimates of heritability and confounding inflation.                                                                                        | R        |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitHub](https://github.com/shuangsong0110/LDER)                      |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GCSC                      | 2022  | [Paper](https://doi.org/10.1016/j.ajhg.2022.01.005)                   | Uses TWAS results in a gene co-regulation score regression, to identify gene sets that are enriched for disease heritability explained by predicted expression.                                                                                      | Python   |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GCSC](https://github.com/ksiewert/GCSC)                              |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GxEsum                    | 2021  | [Paper](https://doi.org/10.1186/s13059-021-02403-1)                   | A method for estimating the phenotypic variance explained by genome-wide GxE.                                                                                                                                                                        | R        |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitHub](https://github.com/honglee0707/GxEsum)                       |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| FMR                       | 2021  | [Paper](https://doi.org/10.1038/s41588-021-00901-3)                   | A method-of-moments estimator of the effect-size distribution. The coefficients quantify the heritability explained by components of a mixture model for the effect-size distribution.                                                               | Matlab   |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitHub](https://github.com/lukejoconnor/FMR)                         |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MESC                      | 2020  | [Paper](https://doi.org/10.1038/s41588-020-0625-2)                    | Estimates the proportion of heritability mediated by assayed gene expression levels using linkage disequilibrium scores and eQTL.                                                                                                                    | Python   |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitHub](https://github.com/douglasyao/mesc)                          |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GWAS-causal-effects-model | 2020  | [Paper](https://doi.org/10.1371/journal.pgen.1008612)                 | Random effects model for estimating the causal variants and their effect size distribution from a dense panel.                                                                                                                                       | Matlab   |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitHub](https://github.com/dominicholland/GWAS-Causal-Effects-Model) |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| SumHer                    | 2019  | [Paper](https://doi.org/10.1038/s41588-018-0279-5)                    | Estimates the SNP heritability of a trait, heritability enrichments and genetic correlations between traits. Part of the LDAK family.                                                                                                                | Exe      |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [Webpage](https://dougspeed.com/sumher/)                              |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GWEHS                     | 2019  | [Paper](https://doi.org/10.3390/genes10080558)                        | Calculates the distribution of effect sizes of SNPs affecting traits, as well as their contribution to heritability. It also allows for predictions as new loci are found.                                                                           | R        |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitLab](https://gitlab.com/elcortegano/GWEHS)                        |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GWIZ                      | 2019  | [Paper](https://doi.org/10.1371/journal.pone.0220215)                 | A method to generate ROC curves and calculate the AUROC.                                                                                                                                                                                             | R        |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitHub](https://github.com/jonaspatronjp/GWIZ-Rscript/)              |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| SummaryAUC                | 2019  | [Paper](https://doi.org/10.1093/bioinformatics/btz176)                | A method for approximating the AUC and its variance of a PRS when only the summary level data of the validation dataset are available.                                                                                                               | R        |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitHub](https://github.com/lsncibb/SummaryAUC)                       |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GENESIS                   | 2018  | [Paper](https://doi.org/10.1038/s41588-018-0193-x)                    | Uses LD information and a likelihood-based approach to estimate variants effect-size distributions. It also allows users to make predictions regarding yield of future GWAS.                                                                         | R        |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitHub](https://github.com/yandorazhang/GENESIS)                     |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| VarExp                    | 2018  | [Paper](https://doi.org/10.1093/bioinformatics/bty379)                | A method that allows for the estimation of the proportion of phenotypic variance explained. It allows for a range of models to be evaluated, including marginal genetic effects, GxE interaction effects and both effects jointly.                   | R        |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitLab](https://gitlab.pasteur.fr/statistical-genetics/VarExp)       |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| S-PCGC                    | 2018  | [Paper](https://doi.org/10.1016/j.ajhg.2018.06.002)                   | An adaptation of stratified LD score regression (S-LDSC) for case-control studies. It can estimate genetic heritability, genetic correlation and functional enrichment.                                                                              | Python   |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitHub](https://github.com/omerwe/S-PCGC)                            |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| HESS                      | 2016  | [Paper](https://doi.org/10.1016/j.ajhg.2016.05.013)                   | Provides utilities for estimating and analyzing local SNP-heritability and genetic covariance.                                                                                                                                                       | Python   |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitHub](https://github.com/huwenboshi/hess)                          |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| S-LDSC                    | 2015  | [Paper](https://doi.org/10.1038/ng.3404)                              | An extension of LDSC for partitioning heritability.                                                                                                                                                                                                  | R        |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitHub](https://github.com/bulik/ldsc)                               |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| LDSC                      | 2015  | [Paper](https://doi.org/10.1038/ng.3211)                              | Distinguishes polygenicity from bias by examining the relationship between test statistics and LD score. Used also for estimating heritability and genetic correlation.                                                                              | Python   |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitHub](https://github.com/bulik/ldsc)                               |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| AVENGEME                  | 2015  | [Paper](https://doi.org/10.1016/j.ajhg.2015.06.005)                   | A method to estimate the variance in disease liability explained by large sets of genetic markers. Uses polygenic scores, based on the formula for the non-centrality parameter of the association test of the score.                                | R        |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitHub](https://github.com/DudbridgeLab/avengeme)                    |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| SumVG                     | 2011  | [Paper](https://doi.org/10.1002/gepi.20593)                           | Provides estimates of the sum of heritability explained by all true susceptibility variants in GWAS. It also estimates the standard error based on re-sampling approaches.                                                                           | R        |
|                           |       |                                                                       |                                                                                                                                                                                                                                                      |          |
|                           |       | [GitHub](https://github.com/lab-hcso/SumVg)                           |                                                                                                                                                                                                                                                      |          |
+---------------------------+-------+-----------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+

: Software are sorted by most to least recent {.striped}

### Gene-based tests

+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| Name           | Year | Links                                                                                                    | Description                                                                                                                                                                                                                                                                                                                                                              | Language |
+================+======+==========================================================================================================+==========================================================================================================================================================================================================================================================================================================================================================================+==========+
| LDAK-GBAT      | 2023 | [Paper](https://doi.org/10.1016/j.ajhg.2022.11.010)                                                      | A computationally efficient method for gene-based association testing. Produces well-calibrated p-values and is significantly more powerful than existing tools. Part of the LDAK family.                                                                                                                                                                                | Exe      |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [Webpage](https://dougspeed.com/ldak-gbat/)                                                              |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| OWC            | 2023 | [Paper](https://doi.org/10.1186/s12859-022-05114-x)                                                      | A gene-based test that incorporates different weighting schemes and includes several popular methods as its special cases (burden test, weighted sum of squared score test \[SSU\], weighted sum statistic \[WSS\], SNP-set Kernel Association Test \[SKAT\], and score test).                                                                                           | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/Xuexia-Wang/OWC-R-package)                                                   |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| PascalX        | 2023 | [Paper](https://doi.org/10.1093/bioinformatics/btad296)                                                  | Provides fast and accurate mapping of SNP-wise GWAS data. It allows for scoring genes and annotated gene sets for enrichment signals based on data from, both, single GWAS and pairs of GWAS.                                                                                                                                                                            | Python   |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/BergmannLab/PascalX)                                                         |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| H-MAGMA        | 2023 | [Paper](https://doi.org/10.1038/s41596-022-00745-z)                                                      | Extends MAGMA by incorporating 3D chromatin configuration in assigning variants to their putative target genes.                                                                                                                                                                                                                                                          | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/thewonlab/H-MAGMA)                                                           |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MCA            | 2022 | [Paper](https://doi.org/10.1186/s12859-022-04897-3)                                                      | Implements 22 different gene-based methods, including linear regression, higher criticism tests, Berk-Jones tests, burden test; SKAT and SKAT-O, Simes and GATES, aggregated Cauchy association test and more.                                                                                                                                                           | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/biostatpzeng/MCA)                                                            |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| EPIC           | 2022 | [Paper](https://doi.org/10.1371/journal.pgen.1010251)                                                    | A method that relates large-scale GWAS summary statistics to cell-type-specific gene expression measurements from single-cell RNA sequencing.                                                                                                                                                                                                                            | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/rujinwang/EPIC)                                                              |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| sumFREGAT      | 2022 | [Paper](https://doi.org/10.1371/journal.pcbi.1010172)                                                    | Offers a wide range of gene-based methods to combine. It allows the user to arbitrarily define a set of these methods, weighting functions and probabilities of genetic variants being causal.                                                                                                                                                                           | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [CRAN](https://cran.r-project.org/web/packages/sumFREGAT/index.html)                                     |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| chromMAGMA     | 2022 | [Paper](https://doi.org/10.26508/lsa.202201446)                                                          | A method to identify candidate risk genes based on the presence of risk variants within noncoding regulatory elements.                                                                                                                                                                                                                                                   | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/lawrenson-lab/chromMAGMA-public)                                             |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| oTFisher       | 2022 | [Paper](https://doi.org/10.3389/fgene.2022.1009428)                                                      | The omnibus thresholding Fisher’s method for performing SNP-set and gene-based tests.                                                                                                                                                                                                                                                                                    | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [CRAN](https://cran.r-project.org/web/packages/TFisher/index.html)                                       |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| HSVS-M         | 2022 | [Paper](https://doi.org/10.1002/gepi.22437)                                                              | A multivariate hierarchically structured variable selection model, a flexible Bayesian model that tests the association of a gene with multiple correlated traits.                                                                                                                                                                                                       | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/yiyangphd/HSVSM)                                                             |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| eMAGMA         | 2021 | [Paper](https://doi.org/10.1093/bioinformatics/btab115)                                                  | A gene-based approach with a modification of MAGMA, leverages significant tissue-specific cis-eQTL information to assign SNPs to putative genes.                                                                                                                                                                                                                         | C/C++    |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/eskederks/eMAGMA-tutorial)                                                   |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| nMAGMA         | 2021 | [Paper](https://doi.org/10.1093/bib/bbaa298)                                                             | An extension of MAGMA which extends the lists of genes that can be annotated to SNPs by integrating local signals, long-range regulation signals, and tissue-specific gene networks. It also provides tissue-specific risk signals, which are useful for understanding disorders with multitissue origins.                                                               | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/sldrcyang/nMAGMA)                                                            |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MARS           | 2021 | [Paper](https://doi.org/10.1186/s13059-021-02353-8)                                                      | Finds associations between variants in risk loci and a phenotype, considering the causal status of variants.                                                                                                                                                                                                                                                             | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/junghyunJJ/marsR)                                                            |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| AgglomerativLD | 2021 | [Paper](https://doi.org/10.1093/bib/bbab329)                                                             | Captures LD of SNPs falling in nearby genes, which induces correlation of gene-based test statistics.                                                                                                                                                                                                                                                                    | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/ryurko/Agglomerative-LD-loci-testing)                                        |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GAMBIT         | 2020 | [Paper](https://doi.org/10.1371/journal.pgen.1009060)                                                    | Integrates heterogeneous annotations with GWAS summary statistics for gene-based analysis, using various coding and tissue-specific regulatory annotations. Allows various tests like SKAT, minP, ACAT, and HMP.                                                                                                                                                         | C/C++    |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/corbinq/GAMBIT)                                                              |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| DOT            | 2020 | [Paper](https://doi.org/10.1371/journal.pcbi.1007819)                                                    | Decorrelation-based approach (DOT) for combining SNP-level summary statistics (or, equivalently, p-values).                                                                                                                                                                                                                                                              | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/dmitri-zaykin/Total_Decor)                                                   |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GBJ            | 2020 | [Paper](https://doi.org/10.1080/01621459.2019.1660170)                                                   | Generalized Berk-Jones test for the association between a SNP-set and outcome by accounting for LD. Includes also tests for Berk-Jones (BJ), Higher Criticism (HC), Generalized Higher Criticism (GHC), Minimum p-value (minP), and an an omnibus test (OMNI).                                                                                                           | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [CRAN](https://cran.r-project.org/web/packages/GBJ/index.html)                                           |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| TS             | 2020 | [Paper](https://doi.org/10.1186/s12859-020-3511-0)                                                       | Uses a truncated method to find the genes that have a true contribution to the genetic association.                                                                                                                                                                                                                                                                      | Exe      |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/Jianjun-CN/c-code-for-TS)                                                    |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| gene-*ε*       | 2020 | [Paper](https://doi.org/10.1371/journal.pgen.1008855)                                                    | A gene-based test using an empirical Bayesian approach and a mixture of normal distributions over the (regularized) effect size estimates.                                                                                                                                                                                                                               | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/ramachandran-lab/genee)                                                      |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GPA            | 2019 | [Paper](https://doi.org/10.1002/gepi.22251)                                                              | A general gene-based p-value adaptive combination approach (GPA) which can integrate association evidence of multiple SNPs. It is applicable to both continuous and binary traits and also to multiple studies.                                                                                                                                                          | C/C++    |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/Biocomputing-Research-Group/GPA)                                             |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| ACAT           | 2019 | [Paper](https://doi.org/10.1016/j.ajhg.2019.01.002)                                                      | A gene-based method using the Cauchy Combination Test. Includes also an omnibus procefure combining SKAT, BT and ACAT.                                                                                                                                                                                                                                                   | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/yaowuliu/ACAT)                                                               |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| KGG            | 2019 | [Paper](https://doi.org/10.1093/bioinformatics/bty682)                                                   | Conditional test that uses a sequential analysis with a linear combination of chi-square statistics. Inherited to KGGSEE.                                                                                                                                                                                                                                                | Java     |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [Webpage](https://pmglab.top/kggsee/#/)                                                                  |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MKATR          | 2018 | [Paper](https://doi.org/10.1016/j.compbiolchem.2018.02.016)                                              | Calculates the correlation of the the test Z-statistics across variants using LD from a population reference panel. Incorporates various tests (sum test, SKAT, adaptive test).                                                                                                                                                                                          | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/baolinwu/mkatr)                                                              |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| FUMA           | 2017 | [Paper](https://doi.org/10.1038/s41467-017-01261-5)                                                      | An integrative web-based platform using information from multiple biological resources to facilitate functional annotation of GWAS results, gene prioritization and interactive visualization. It accommodates positional, expression quantitative trait loci (eQTL) and chromatin interaction mappings, and provides gene-based, pathway and tissue enrichment results. | web      |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [Webpage](https://fuma.ctglab.nl/)                                                                       |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| aSPU           | 2017 | [Paper](https://doi.org/10.1093/bioinformatics/btw577)                                                   | Performs adaptive gene-based test and pathway-based test for association analysis of multiple traits. The tests are adaptive at both the SNP- and trait-levels, thus maintaining high power across a wide range of situations. The methods can be applied to mixed types of traits, and to z-statistics or p-values.                                                     | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/ikwak2/aSPU)                                                                 |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| COMBAT         | 2017 | [Paper](https://doi.org/10.1534/genetics.117.300257)                                                     | A combined association test for genes, which incorporates strengths from existing gene-based tests and shows higher overall performance than individual tests.                                                                                                                                                                                                           | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [CRAN](https://cran.r-project.org/web/packages/COMBAT/)                                                  |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| FSTpackage     | 2017 | [Paper](https://doi.org/10.1016/j.ajhg.2017.07.011)                                                      | Combines dispersion and burden tests and an efficient perturbation method for individual gene/large gene-set/genome wide analysis.                                                                                                                                                                                                                                       | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [CRAN](https://cran.r-project.org/web/packages/FSTpackage/index.html)                                    |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| snpGeneSets    | 2016 | [Paper](https://doi.org/10.1534/g3.116.034694)                                                           | Integrates local genomic annotation databases and provides genome-wide annotation for SNP, Gene and gene sets. Aims to support interpretation of GWAS results and performing post-analysis.                                                                                                                                                                              | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [Webpage](https://www.umc.edu/SoPH/Departments-and-Faculty/Data-Science/Research/Services/Software.html) |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| PASCAL         | 2016 | [Paper](https://doi.org/10.1371/journal.pcbi.1004714)                                                    | Computes gene and pathway scores from SNP-phenotype associations. For gene score computation, implements analytic and efficient numerical solutions to calculate test statistics. For pathway scoring, uses a modified Fisher method.                                                                                                                                    | Java     |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [Webpage](https://www2.unil.ch/cbg/index.php?title=Pascal)                                               |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| JEPEGMIX       | 2016 | [Paper](https://doi.org/10.1093/bioinformatics/btv567)                                                   | Extends JEPEG to better model mixed ethnicity cohorts.                                                                                                                                                                                                                                                                                                                   | Exe      |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://dleelab.github.io/jepegmix/)                                                            |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| PEGASUS        | 2016 | [Paper](https://doi.org/10.1534/genetics.116.188391)                                                     | A gene-based method that uses an analytical approach to compute gene-level p-values of observed gene scores according to a null distribution modeling LD.                                                                                                                                                                                                                | Perl     |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/ramachandran-lab/PEGASUS)                                                    |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| fastBAT        | 2016 | [Paper](https://doi.org/10.1038/srep32894)                                                               | Performs a fast set-based association analysis for human complex traits using summary-level data from GWAS and LD. Part of the GCTA family.                                                                                                                                                                                                                              | C/C++    |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [Webpage](https://yanglab.westlake.edu.cn/software/gcta/#fastBAT)                                        |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MAGMA          | 2015 | [Paper](https://doi.org/10.1371/journal.pcbi.1004219)                                                    | Uses p-values and performs gene-based and gene-set analysis as well as meta-analysis.                                                                                                                                                                                                                                                                                    | C/C++    |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [Webpage](https://cncr.nl/research/magma/)                                                               |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| JEPEG          | 2015 | [Paper](https://doi.org/10.1093/bioinformatics/btu816)                                                   | A gene-based method for testing the joint effects on trait for SNPs functionally associated with a gene (eQTLs).                                                                                                                                                                                                                                                         | Exe      |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](http://dleelab.github.io/jepeg/)                                                                |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GCTA-COJO      | 2012 | [Paper](https://doi.org/10.1038/ng.2213)                                                                 | An approximate conditional and joint association analysis that uses LD from a reference sample. Part of the GCTA family.                                                                                                                                                                                                                                                 | C/C++    |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [Webpage](https://yanglab.westlake.edu.cn/software/gcta/#COJO)                                           |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GATES          | 2011 | [Paper](https://doi.org/10.1016/j.ajhg.2011.01.019)                                                      | An extended Sime's test that integrates functional information and association evidence to combine the p-values of the SNP within a gene to obtain an overall p-value. Inherited to KGGSEE.                                                                                                                                                                              | Java     |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [Webpage](https://pmglab.top/kggsee/#/)                                                                  |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| VEGAS          | 2010 | [Paper](https://doi.org/10.1016/j.ajhg.2010.06.009)                                                      | One of the first multivariate methods. Takes account of LD between markers in a gene by using simulation based on the LD of a reference panel.                                                                                                                                                                                                                           | R        |
|                |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |          |
|                |      | [GitHub](https://github.com/HimesGroup/snpsettest)                                                       |                                                                                                                                                                                                                                                                                                                                                                          |          |
+----------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+

: Software are sorted by most to least recent {.striped}

### Gene set analysis

+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| Name             | Year | Links                                                                                                    | Description                                                                                                                                                                                                                                                                                                                                                              | Language   |
+==================+======+==========================================================================================================+==========================================================================================================================================================================================================================================================================================================================================================================+============+
| PascalX          | 2023 | [Paper](https://doi.org/10.1093/bioinformatics/btad296)                                                  | Provides fast and accurate mapping of SNP-wise GWAS data. It allows for scoring genes and annotated gene sets for enrichment signals based on data from, both, single GWAS and pairs of GWAS.                                                                                                                                                                            | Python     |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [GitHub](https://github.com/BergmannLab/PascalX)                                                         |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| g:Profiler       | 2023 | [Paper](https://doi.org/10.1093/nar/gkad347)                                                             | Integrates many databases, including Gene Ontology, KEGG and TRANSFAC, to provide a comprehensive and in-depth analysis of gene lists. It also provides interactive and intuitive user interfaces and supports ordered queries and custom statistical backgrounds, among other settings.                                                                                 | R/web      |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Webpage](https://biit.cs.ut.ee/gprofiler/gost)                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| DAVID            | 2022 | [Paper](https://doi.org/10.1093/nar/gkac194)                                                             | An enrichment tool with functionalities for different types of omics data including GWAS. It accepts gene or SNP-list as input and provide API ensuring interoperability. For analysis it uses ORA and GSEA.                                                                                                                                                             | R/web      |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Webpage](https://davidbioinformatics.nih.gov/)                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| chromMAGMA       | 2022 | [Paper](https://doi.org/10.26508/lsa.202201446)                                                          | A method to identify candidate risk genes based on the presence of risk variants within noncoding regulatory elements.                                                                                                                                                                                                                                                   | R          |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [GitHub](https://github.com/lawrenson-lab/chromMAGMA-public)                                             |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| PANTHER (16)     | 2021 | [Paper](https://doi.org/10.1093/nar/gkaa1106)                                                            | An enrichment tool with functionalities for different types of omics data including GWAS. It accepts gene or SNP-list as input and provide API ensuring interoperability. For analysis it uses ORA and GSEA.                                                                                                                                                             | web        |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Webpage](https://www.pantherdb.org/)                                                                    |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| Mergeomics (2.0) | 2021 | [Paper](https://doi.org/10.1093/nar/gkab405)                                                             | A web server which uses summary statistics of multi-omics association studies (GWAS, EWAS, TWAS, PWAS, etc) and performs correction for LD, GSEA, meta-analysis and identification of essential regulators of disease-associated pathways and networks.                                                                                                                  | R/web      |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Webpage](https://mergeomics.research.idre.ucla.edu/)                                                    |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| GAUSS            | 2021 | [Paper](https://doi.org/10.1016/j.ajhg.2021.02.016)                                                      | Tests for any self-contained association between a phenotype and a gene-set and produces a p-value for the association.                                                                                                                                                                                                                                                  | R          |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [GitHub](https://github.com/diptavo/GAUSS)                                                               |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| GEMB             | 2020 | [Paper](https://doi.org/10.1093/gigascience/giaa091)                                                     | A method that combines gene weights from model predictions and gene ranks from genome-wide association studies into a weighted gene-set test.                                                                                                                                                                                                                            | Matlab     |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [GitHub](https://github.com/cochran4/GEMB)                                                               |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| GIGSEA           | 2019 | [Paper](https://doi.org/10.1093/bioinformatics/bty529)                                                   | Uses GWAS and eQTL to infer differential gene expression and interrogate gene set enrichment for the trait-associated SNPs. By incorporating expression data it naturally accounts for factors such as gene size, gene boundary, SNP distal regulation and multiple-marker regulation.                                                                                   | R          |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [GitHub](https://github.com/zhushijia/GIGSEA)                                                            |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| WebGestalt       | 2019 | [Paper](https://doi.org/10.1093/nar/gkz401)                                                              | An enrichment tool with functionalities for different types of omics data including GWAS. It accepts gene or SNP-list as input and provide API ensuring interoperability. For analysis it uses ORA, GSEA and Network Topology-based Analysis.                                                                                                                            | R/web      |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Webpage](https://www.webgestalt.org/)                                                                   |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| deTS             | 2019 | [Paper](https://doi.org/10.1093/bioinformatics/btz138)                                                   | Performs tissue-specific enrichment analysis (TSEA) for detecting tissue-specific genes and for enrichment test of different forms of query data.                                                                                                                                                                                                                        | R          |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [GitHub](https://github.com/bsml320/deTS)                                                                |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| DESE             | 2019 | [Paper](https://doi.org/10.1186/s13059-019-1801-5)                                                       | Detects the causal tissues of complex traits according to selective expression of disease-associated genes.                                                                                                                                                                                                                                                              | web        |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Webpage](https://pmglab.top/pcga/#/)                                                                    |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| INFERNO          | 2018 | [Paper](https://doi.org/10.1093/nar/gky686)                                                              | A method which integrates diverse functional genomics data sources to identify causal noncoding variants. Characterizes the relevant tissue contexts, target genes, and downstream biological processes affected by functional variants. Uses COLOC, WebGestalt, LDSC and MetaXcan.                                                                                      | Python/web |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Webpage](http://inferno.lisanwanglab.org/index.php)                                                     |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| aSPUpath2        | 2018 | [Paper](https://doi.org/10.1002/gepi.22110)                                                              | Integrates gene expression reference weights, GWAS summary data, LD information, and candidate pathways to identify pathways whose expression is associated with complex traits.                                                                                                                                                                                         | R          |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Webpage](https://wuchong.org/aspupath2.html)                                                            |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| GSA-SNP2         | 2018 | [Paper](https://doi.org/10.1093/nar/gky175)                                                              | A method for pathway enrichment analysis of GWAS P-value data. It accepts also gene-wise p-values (obtained from other methods) and outputs pathway gene sets ‘enriched’ with genes associated with the given phenotype.                                                                                                                                                 | C/C++      |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Forge](https://sourceforge.net/projects/gsasnp2/)                                                       |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| FUMA             | 2017 | [Paper](https://doi.org/10.1038/s41467-017-01261-5)                                                      | An integrative web-based platform using information from multiple biological resources to facilitate functional annotation of GWAS results, gene prioritization and interactive visualization. It accommodates positional, expression quantitative trait loci (eQTL) and chromatin interaction mappings, and provides gene-based, pathway and tissue enrichment results. | web        |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Webpage](https://fuma.ctglab.nl/)                                                                       |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| GWAB             | 2017 | [Paper](https://doi.org/10.1093/nar/gkx284)                                                              | Trait-associated genes with sub-threshold significance score can be rescued by network connections to other significant candidates.                                                                                                                                                                                                                                      | web        |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Webpage](https://www.inetbio.org/gwab/)                                                                 |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| aSPU             | 2017 | [Paper](https://doi.org/10.1093/bioinformatics/btw577)                                                   | Performs adaptive gene-based test and pathway-based test for association analysis of multiple traits. The tests are adaptive at both the SNP- and trait-levels, thus maintaining high power across a wide range of situations. The methods can be applied to mixed types of traits, and to z-statistics or p-values.                                                     | R          |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [GitHub](https://github.com/ikwak2/aSPU)                                                                 |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| snpGeneSets      | 2016 | [Paper](https://doi.org/10.1534/g3.116.034694)                                                           | Integrates local genomic annotation databases and provides genome-wide annotation for SNP, Gene and gene sets. Aims to support interpretation of GWAS results and performing post-analysis.                                                                                                                                                                              | R          |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Webpage](https://www.umc.edu/SoPH/Departments-and-Faculty/Data-Science/Research/Services/Software.html) |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| PASCAL           | 2016 | [Paper](https://doi.org/10.1371/journal.pcbi.1004714)                                                    | Computes gene and pathway scores from SNP-phenotype associations. For gene score computation, implements analytic and efficient numerical solutions to calculate test statistics. For pathway scoring, uses a modified Fisher method.                                                                                                                                    | Java       |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Webpage](https://www2.unil.ch/cbg/index.php?title=Pascal)                                               |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| Enrichr          | 2016 | [Paper](https://doi.org/10.1093/nar/gkw377)                                                              | A gene set search engine that enables the querying of hundreds of thousands of annotated gene sets. Enrichr uniquely integrates knowledge from many high-profile projects to provide synthesized information about mammalian genes and gene sets.                                                                                                                        | web        |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Webpage](https://maayanlab.cloud/Enrichr/)                                                              |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| PAPA             | 2016 | [Paper](https://doi.org/10.1093/bioinformatics/btv668)                                                   | A flexible tool for pleiotropic pathway analysis utilizing GWAS summary results.                                                                                                                                                                                                                                                                                         | C/C++      |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Forge](https://sourceforge.net/projects/papav1/files/)                                                  |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| GenToS           | 2016 | [Paper](https://doi.org/10.1371/journal.pone.0162466)                                                    | Calculates an appropriate statistical significance threshold and then searches for trait-associated variants in summary statistics from human GWAS.                                                                                                                                                                                                                      | Java       |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [GitHub](https://github.com/genepi-freiburg/gentos)                                                      |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| MAGMA            | 2015 | [Paper](https://doi.org/10.1371/journal.pcbi.1004219)                                                    | Uses p-values and performs gene-based and gene-set analysis as well as meta-analysis.                                                                                                                                                                                                                                                                                    | C/C++      |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Webpage](https://cncr.nl/research/magma/)                                                               |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| VSEAMS           | 2014 | [Paper](https://doi.org/10.1093/bioinformatics/btu571)                                                   | A non-parametric SNP set enrichment method to test for enrichment of GWAS signals in functionally defined loci using p-values.                                                                                                                                                                                                                                           | E          |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [GitHub](https://github.com/ollyburren/vseams)                                                           |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| GENOMICper       | 2012 | [Paper](https://doi.org/10.1534/g3.112.002618)                                                           | Uses SNP association p-values and permutes them by rotation with respect to the genomic locations. The joint gene p-values are calculated using Fisher’s combination test and pathways’ association tested using the hypergeometric test.                                                                                                                                | R          |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [CRAN](https://cran.r-project.org/web/packages/genomicper/index.html)                                    |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| dmGWAS           | 2011 | [Paper](https://doi.org/10.1093/bioinformatics/btq615)                                                   | A dense module searching method to identify candidate subnetworks or genes for complex diseases by integrating PPI network. Extensively searches for subnetworks enriched with low p-value genes.                                                                                                                                                                        | Exe        |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Webpage](https://bioinfo.uth.edu/dmGWAS/)                                                               |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| i-GSEA4GWAS      | 2010 | [Paper](https://doi.org/10.1093/nar/gkq324)                                                              | Detects pathways associated with traits by applying an improved gene set enrichment analysis. Implements also a follow-up functional analysis for SNPs in trait-associated pathways identified. Uses LD and putative functional annotation from Ensembl,ENCODE and eQTLs.                                                                                                | web        |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Webpage](http://gsea4gwas.psych.ac.cn/GSEA4GWAS/)                                                       |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| SNPratio test    | 2009 | [Paper](https://doi.org/10.1093/bioinformatics/btp448)                                                   | Compares the proportion of significant to all SNPs within genes that are part of a pathway and computes an empirical p-value based on comparisons to ratios in datasets where the assignment of case/control status has been randomized.                                                                                                                                 | Perl       |
|                  |      |                                                                                                          |                                                                                                                                                                                                                                                                                                                                                                          |            |
|                  |      | [Forge](https://sourceforge.net/projects/snpratiotest/)                                                  |                                                                                                                                                                                                                                                                                                                                                                          |            |
+------------------+------+----------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+

: Software are sorted by most to least recent {.striped}

### Fine-mapping

+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| Name            | Year     | Links                                                                                                                               | Description                                                                                                                                                                                                                | Language |
+=================+==========+=====================================================================================================================================+============================================================================================================================================================================================================================+==========+
| BEATRICE        | Preprint | [Paper](https://doi.org/10.1101/2023.03.24.534116)                                                                                  | Combines a hierarchical Bayesian model with a deep learning-based inference procedure.                                                                                                                                     | Python   |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://github.com/sayangsep/Beatrice-Finemapping)                                                                         |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| XMAP            | 2023     | [Paper](https://doi.org/10.1038/s41467-023-42614-7)                                                                                 | A variational EM method for cross-population fine-mapping by leveraging genetic diversity and accounting for confounding bias.                                                                                             | R        |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://github.com/YangLabHKUST/XMAP)                                                                                      |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| finiMOM         | 2023     | [Paper](https://doi.org/10.1093/bioinformatics/btad396)                                                                             | A Bayesian method that allows for multiple causal variants using product inverse-moment prior which is a natural prior distribution to model non-null effects in finite samples.                                           | R        |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://vkarhune.github.io/finimom/)                                                                                       |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| CARMA           | 2023     | [Paper](https://doi.org/10.1038/s41588-023-01392-0)                                                                                 | Bayesian model that allows flexible specification of the prior distribution, joint modeling of summary statistics and functional annotations, and accounting for discrepancies between summary statistics and external LD. | R        |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://github.com/Iuliana-Ionita-Laza/CARMA)                                                                              |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| SusieR          | 2022     | [Paper](https://doi.org/10.1371/journal.pgen.1010299)                                                                               | Performs variable selection in multiple regression using a Bayesian version of stepwise selection approach, and is particularly well-suited to settings where some of the variables are highly correlated.                 | R        |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://github.com/stephenslab/susieR)                                                                                     |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| echocolatoR     | 2022     | [Paper](https://doi.org/10.1093/bioinformatics/btab658)                                                                             | Integrates a diverse suite of statistical and functional fine-mapping tools to identify, test enrichment in, and visualize high-confidence causal consensus variants in any phenotype. Part of the echoverse family.       | R        |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://rajlabmssm.github.io/echolocatoR/)                                                                                 |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| PICS2           | 2021     | [Paper](https://doi.org/10.1093/bioinformatics/btab122)                                                                             | A fine-mapping tool for determining the likelihood that each SNP in LD with a reported index SNP is a true causal polymorphism.                                                                                            | web      |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [Webpage](https://pics2.ucsf.edu/)                                                                                                  |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| ANNORE          | 2021     | [Paper](https://doi.org/10.1093/hmg/ddab210)                                                                                        | Uses local LD structure and functional annotation, accross many categories, to prioritize the most plausible causal SNPs. It is based on multiple regression with differential shrinkage via random effects.               | R        |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://github.com/vafisher/AnnoRE)                                                                                        |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| flashfm         | 2021     | [Paper](https://doi.org/10.1038/s41467-021-26364-y)                                                                                 | Uses summary statistics to jointly fine-map genetic associations for several related quantitative traits in a Bayesian framework that leverages information between the traits.                                            | R        |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://jennasimit.github.io/flashfm/)                                                                                     |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MsCAVIAR        | 2021     | [Paper](https://doi.org/10.1371/journal.pgen.1009733)                                                                               | A method for fine-mapping by leveraging information from multiple studies. One important application area is trans-ethnic fine mapping.                                                                                    | C/C++    |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://github.com/nlapier2/MsCAVIAR)                                                                                      |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| JOINTSUM        | 2020     | [Paper](https://doi.org/10.1371/journal.pcbi.1007778)                                                                               | A simple and general approach based on conditional analysis of a locus on multiple traits, overcoming the shortcomings of other methods.                                                                                   | R        |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://github.com/yangq001/conditional)                                                                                   |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GSR             | 2020     | [Paper](https://doi.org/10.1371/journal.pone.0237657)                                                                               | Detects causal gene sets for complex traits using gene score regression while accounting for gene-to-gene correlations. It can operate either on GWAS summary statistics or gene expression.                               | Python   |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://github.com/li-lab-mcgill/GSR)                                                                                      |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| PolyFun/PolyLoc | 2020     | [Paper](https://doi.org/10.1038/s41588-020-00735-5)                                                                                 | Estimates prior causal probabilities for SNPs, which can then be used by fine-mapping methods like SuSiE or FINEMAP. It can aggregate polygenic data from across the entire genome and hundreds of functional annotations. | Python   |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://github.com/omerwe/polyfun)                                                                                         |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| AHIUT           | 2018     | [Paper](https://doi.org/10.1534/genetics.118.301111)                                                                                | An intersection-union test based on a joint/conditional regression model with all the SNPs in a locus to infer AH.                                                                                                         | R        |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [Figshare](https://figshare.com/articles/dataset/AHIUT/6615470?file=12108314)                                                       |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| PAINTOR         | 2017     | [Paper](https://doi.org/10.1093/bioinformatics/btw615)                                                                              | Integrates functional genomic data with association strength from potentially multiple populations (or traits) to prioritize variants for follow-up analysis.                                                              | C/C++    |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://github.com/gkichaev/PAINTOR_V3.0)                                                                                  |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| SOJO            | 2017     | [Paper](https://doi.org/10.1016/j.ajhg.2017.09.027)                                                                                 | Penalized selection operator for jointly analyzing multiple variants (SOJO) within each mapped locus on the basis of LASSO regression derived from summary association statistics.                                         | R        |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://github.com/zhenin/sojo)                                                                                            |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| HAPRAP          | 2017     | [Paper](https://doi.org/10.1093/bioinformatics/btw565)                                                                              | An empirical iterative method, that enables fine mapping using haplotype information from an individual-level reference panel.                                                                                             | Python   |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [Biocompute](http://apps.biocompute.org.uk/haprap/)                                                                                 |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| RSS             | 2017     | [Paper](https://doi.org/10.1214/17-aoas1046)                                                                                        | Uses a “Regression with Summary Statistics” (RSS) likelihood, which relates the multiple regression coefficients to univariate regression results.                                                                         | Matlab   |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://stephenslab.github.io/rss/)                                                                                        |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| BVS-PICA        | 2017     | [Paper](https://doi.org/10.1111/biom.12620)                                                                                         | Bayesian variable selection for classifying genomic class level associations.                                                                                                                                              | R        |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [Download](https://onlinelibrary.wiley.com/action/downloadSupplement?doi=10.1111%2Fbiom.12620&file=biom12620-sup-0001-SuppData.zip) |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| FINEMAP         | 2016     | [Paper](https://doi.org/10.1093/bioinformatics/btw018)                                                                              | Applies a shotgun stochastic search algorithm and can identify causal SNPs, estimate their effect sizes and the heritability contribution of causal SNPs in genomic regions associated with complex traits.                | Exe      |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [Webpage](http://www.christianbenner.com/)                                                                                          |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| RiVERIA-beta    | 2016     | [Paper](https://doi.org/10.1093/nar/gkw627)                                                                                         | Bayesian fine-mapping using Epigenomic Reference Annotation.                                                                                                                                                               | R        |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://yueli-compbio.github.io/RiVIERA-beta/)                                                                             |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| DAP             | 2016     | [Paper](https://doi.org/10.1016/j.ajhg.2016.03.029)                                                                                 | Deterministic approximation of posteriors enables highly efficient and accurate joint enrichment analysis and identification of multiple causal variants.                                                                  | C/C++    |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://github.com/xqwen/dap)                                                                                              |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| JAM             | 2016     | [Paper](https://doi.org/10.1002/gepi.21953)                                                                                         | Bayesian penalized regression that accounts for SNP correlation and finds SNPs that best explain the complete joint pattern of marginal effects.                                                                           | R        |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://github.com/pjnewcombe/R2BGLiMS)                                                                                    |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| CAVIARBF        | 2015     | [Paper](https://doi.org/10.1534/genetics.115.176107)                                                                                | A fine-mapping method that combines CAVIAR with Bayesian inference using marginal test statistics.                                                                                                                         | C/C++    |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [Bitbucket](https://bitbucket.org/Wenan/caviarbf/src/master/)                                                                       |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| fgwas           | 2014     | [Paper](https://doi.org/10.1016/j.ajhg.2014.03.004)                                                                                 | Integrates functional genomic information into a GWAS.                                                                                                                                                                     | C/C++    |
|                 |          |                                                                                                                                     |                                                                                                                                                                                                                            |          |
|                 |          | [GitHub](https://github.com/joepickrell/fgwas)                                                                                      |                                                                                                                                                                                                                            |          |
+-----------------+----------+-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+

: Software are sorted by most to least recent {.striped}

## Multiple trait analysis

### Genetic correlation

+------------+------+--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| Name       | Year | Links                                                  | Description                                                                                                                                                                                                                        | Language |
+============+======+========================================================+====================================================================================================================================================================================================================================+==========+
| LAVA       | 2022 | [Paper](https://doi.org/10.1038/s41588-022-01017-y)    | An integrated framework for local genetic correlation analysis that can also evaluate local heritabilities and analyze conditional genetic relations between several phenotypes using partial correlation and multiple regression. | R        |
|            |      |                                                        |                                                                                                                                                                                                                                    |          |
|            |      | [GitHub](https://github.com/josefin-werme/LAVA)        |                                                                                                                                                                                                                                    |          |
+------------+------+--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| LOGOdetect | 2021 | [Paper](https://doi.org/10.1038/s41467-021-22334-6)    | A tool to identify small segments that harbor local genetic correlation between two traits.                                                                                                                                        | R        |
|            |      |                                                        |                                                                                                                                                                                                                                    |          |
|            |      | [GitHub](https://github.com/ghm17/LOGODetect)          |                                                                                                                                                                                                                                    |          |
+------------+------+--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| CC-GWAS    | 2021 | [Paper](https://doi.org/10.1038/s41588-021-00787-1)    | A tool for case-case association testing of two different disorders.                                                                                                                                                               | R        |
|            |      |                                                        |                                                                                                                                                                                                                                    |          |
|            |      | [GitHub](https://github.com/wouterpeyrot/CCGWAS)       |                                                                                                                                                                                                                                    |          |
+------------+------+--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| DONUTS     | 2021 | [Paper](https://doi.org/10.1073/pnas.2023184118)       | A statistical framework that can estimate direct and indirect genetic effects at the SNP level and calculate genetic correlation between traits.                                                                                   | R        |
|            |      |                                                        |                                                                                                                                                                                                                                    |          |
|            |      | [GitHub](https://github.com/qlu-lab/DONUTS)            |                                                                                                                                                                                                                                    |          |
+------------+------+--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| SUPERGNOVA | 2021 | [Paper](https://doi.org/10.1186/s13059-021-02478-w)    | Extends GNOVA to identify global and local genetic correlations that could provide new insights into the shared genetic basis of many phenotypes.                                                                                  | Python   |
|            |      |                                                        |                                                                                                                                                                                                                                    |          |
|            |      | [GitHub](https://github.com/qlu-lab/SUPERGNOVA)        |                                                                                                                                                                                                                                    |          |
+------------+------+--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GECKO      | 2021 | [Paper](https://doi.org/10.1371/journal.pgen.1009293)  | A method based on composite likelihood for estimating genetic and environmental covariances.                                                                                                                                       | R        |
|            |      |                                                        |                                                                                                                                                                                                                                    |          |
|            |      | [GitHub](https://github.com/borangao/GECKO)            |                                                                                                                                                                                                                                    |          |
+------------+------+--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| LPM        | 2020 | [Paper](https://doi.org/10.1093/bioinformatics/btz947) | A latent probit model that can integrate functional annotations. It is scalable to hundreds of annotations and phenotypes.                                                                                                         | R        |
|            |      |                                                        |                                                                                                                                                                                                                                    |          |
|            |      | [GitHub](https://github.com/mingjingsi/LPM)            |                                                                                                                                                                                                                                    |          |
+------------+------+--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| HDL        | 2020 | [Paper](https://doi.org/10.1038/s41588-020-0653-y)     | A likelihood-based method for estimating genetic correlation. Compared to LDSC, It reduces the variance of a genetic correlation estimate by about 60%.                                                                            | R        |
|            |      |                                                        |                                                                                                                                                                                                                                    |          |
|            |      | [GitHub](https://github.com/zhenin/HDL)                |                                                                                                                                                                                                                                    |          |
+------------+------+--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| SumHer     | 2019 | [Paper](https://doi.org/10.1038/s41588-018-0279-5)     | Estimates the SNP heritability of a trait, heritability enrichments and genetic correlations between traits. Part of the LDAK family.                                                                                              | Exe      |
|            |      |                                                        |                                                                                                                                                                                                                                    |          |
|            |      | [Webpage](https://dougspeed.com/sumher/)               |                                                                                                                                                                                                                                    |          |
+------------+------+--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| S-PCGC     | 2018 | [Paper](https://doi.org/10.1016/j.ajhg.2018.06.002)    | An adaptation of stratified LD score regression (S-LDSC) for case-control studies. It can estimate genetic heritability, genetic correlation and functional enrichment.                                                            | Python   |
|            |      |                                                        |                                                                                                                                                                                                                                    |          |
|            |      | [GitHub](https://github.com/omerwe/S-PCGC)             |                                                                                                                                                                                                                                    |          |
+------------+------+--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| PhenoSpD   | 2018 | [Paper](https://doi.org/10.1093/gigascience/giy090)    | Uses LDSC to estimate phenotypic correlations and then performs correction of multiple testing using the spectral decomposition of matrices.                                                                                       | R        |
|            |      |                                                        |                                                                                                                                                                                                                                    |          |
|            |      | [GitHub](https://github.com/MRCIEU/PhenoSpD)           |                                                                                                                                                                                                                                    |          |
+------------+------+--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| RHOGE      | 2017 | [Paper](https://doi.org/10.1016/j.ajhg.2017.01.031)    | Estimates the genetic correlation between two complex traits as a function of predicted gene expression effect.                                                                                                                    | R        |
|            |      |                                                        |                                                                                                                                                                                                                                    |          |
|            |      | [GitHub](https://github.com/bogdanlab/RHOGE)           |                                                                                                                                                                                                                                    |          |
+------------+------+--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GNOVA      | 2017 | [Paper](https://doi.org/10.1016/j.ajhg.2017.11.001)    | A method that calculates annotation-stratified covariance between arbitrary number of traits and enables researchers to dissect both the shared and distinct genetic architecture across traits.                                   | Python   |
|            |      |                                                        |                                                                                                                                                                                                                                    |          |
|            |      | [GitHub](https://github.com/xtonyjiang/GNOVA)          |                                                                                                                                                                                                                                    |          |
+------------+------+--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| HESS       | 2016 | [Paper](https://doi.org/10.1016/j.ajhg.2017.09.022)    | Provides utilities for estimating and analyzing local SNP-heritability and genetic covariance.                                                                                                                                     | Python   |
|            |      |                                                        |                                                                                                                                                                                                                                    |          |
|            |      | [GitHub](https://github.com/huwenboshi/hess)           |                                                                                                                                                                                                                                    |          |
+------------+------+--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| Popcorn    | 2016 | [Paper](https://doi.org/10.1016/j.ajhg.2016.05.001)    | A method for estimating the transethnic genetic correlation: the correlation of causal-variant effect sizes at SNPs common in populations.                                                                                         | Python   |
|            |      |                                                        |                                                                                                                                                                                                                                    |          |
|            |      | [GitHub](https://github.com/brielin/Popcorn)           |                                                                                                                                                                                                                                    |          |
+------------+------+--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| LDSC       | 2015 | [Paper](https://doi.org/10.1038/ng.3406)               | Distinguishes polygenicity from bias by examining the relationship between test statistics and LD score. Used also for estimating heritability and genetic correlation.                                                            | Python   |
|            |      |                                                        |                                                                                                                                                                                                                                    |          |
|            |      | [GitHub](https://github.com/bulik/ldsc)                |                                                                                                                                                                                                                                    |          |
+------------+------+--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+

: Software are sorted by most to least recent {.striped}

### Pleiotropy

+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| Name            | Year | Links                                                                                | Description                                                                                                                                                                                                                                                                            | Language   |
+=================+======+======================================================================================+========================================================================================================================================================================================================================================================================================+============+
| sumDAG          | 2024 | [Paper](https://doi.org/10.1093/biomtc/ujad039)                                      | Constructs a phenotype network by assuming a Gaussian linear structure model embedding a directed acyclic graph.                                                                                                                                                                       | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/chunlinli/sumdag)                                        |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| graph-GPA (2.0) | 2023 | [Paper](https://doi.org/10.3389/fgene.2023.1079198)                                  | Bayesian graphical model which allows to integrate functional annotations with GWAS datasets for multiple phenotypes within a unified framework.                                                                                                                                       | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://dongjunchung.github.io/GGPA2/)                                      |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| GCPBayes        | 2023 | [Paper](https://doi.org/10.1093/nargab/lqad065)                                      | A pipeline to perform cross-phenotype gene-set analysis between two traits using GCPBayes.                                                                                                                                                                                             | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/CESP-ExpHer/GCPBayes-Pipeline)                           |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| FactorGo        | 2023 | [Paper](https://doi.org/10.1016/j.ajhg.2023.09.015)                                  | A scalable variational factor analysis model used to identify and characterize pleiotropic components. Works well in capturing latent pleiotropic factors across phenotypes while at the same time being computationally efficient.                                                    | Python     |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/mancusolab/FactorGo)                                     |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| MTAFS           | 2023 | [Paper](https://doi.org/10.1038/s41431-023-01389-7)                                  | An efficient and robust adaptive method for multi-trait analysis of GWAS.                                                                                                                                                                                                              | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/Qiaolan/MTAFS)                                           |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| TWT             | 2023 | [Paper](https://doi.org/10.1093/bioinformatics/btad182)                              | Uses the correlation coefficients between Wald statistics obtained from linear regression with covariates. Then, a test is applied by integrating three-level information including the intrinsic genetic structure, pleiotropy, and the potential information combinations.           | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/bschilder/ThreeWayTest)                                  |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| SHAHER          | 2022 | [Paper](https://doi.org/10.3390/genes13101694)                                       | It is based on the construction of a linear combination of traits by maximizing the proportion of its genetic variance explained by the shared genetic factors.                                                                                                                        | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/Sodbo/shared_heredity)                                   |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| PAT             | 2022 | [Paper](https://doi.org/10.1371/journal.pgen.1010447)                                | The pleiotropic association test (PAT) is used for joint analysis of multiple traits. Uses the decomposition of phenotypic covariation into genetic and environmental components to create a likelihood ratio test statistic for each genetic variant.                                 | Python     |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/koditaraszka/pat)                                        |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| EBMMT           | 2022 | [Paper](https://doi.org/10.1002/gepi.22439)                                          | Uses the eigen higher criticism and the eigen Berk-Jones testing procedures to test the association between SNPs and multiple correlated traits. Then uses the aggregated Cauchy association test.                                                                                     | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/Vivian-Liu-Wei64/EBMMT)                                  |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| PDR             | 2022 | [Paper](https://doi.org/10.1016/j.ajhg.2022.04.003)                                  | Pleiotropic decomposition regression using method of moments to identify shared components and their underlying genetic variants.                                                                                                                                                      | Matlab     |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/jballard28/PDR)                                          |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| MAIUP           | 2022 | [Paper](https://doi.org/10.1093/bib/bbab389)                                         | Test constructed based on the traditional intersection–union test with two sets of independent p-values as input and follows a novel idea that was originally proposed under the high-dimensional mediation analysis framework.                                                        | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/biostatpzeng/MAIUP)                                      |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| PolarMorphism   | 2022 | [Paper](https://doi.org/10.1093/bioinformatics/btac228)                              | A method based on a transform from Cartesian to polar coordinates. Analyzes multiple related phenotypes and reports (per SNP) the degree of 'sharedness' across them, its overall effect size, as well as p-values.                                                                    | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/UMCUGenetics/PolarMorphism)                              |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| PLEIO           | 2021 | [Paper](https://doi.org/10.1016/j.ajhg.2020.11.017)                                  | A framework to map and interpret pleiotropic loci in a joint analysis of multiple diseases and complex traits. It maximizes power by systematically accounting for genetic correlations and heritability of the traits using LDSC.                                                     | Python     |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/cuelee/pleio)                                            |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| ACA             | 2021 | [Paper](https://doi.org/10.1038/s41598-021-82000-1)                                  | Relies on an approximate conditional phenotype analysis. The traits covariance may be estimated either from a subset of the phenotypic data; or from published studies.                                                                                                                | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [Webpage](https://sites.bu.edu/fhspl/publications/approximate-conditional-analysis/) |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| combGWAS        | 2021 | [Paper](https://doi.org/10.1093/bioinformatics/btab417)                              | A statistical framework to uncover susceptibility variants for comorbid disorders and calculate genetic correlations.                                                                                                                                                                  | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/LiangyingYin/CombGWAS)                                   |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| jaSPU           | 2021 | [Paper](https://doi.org/10.1093/hmg/ddab126)                                         | Evaluates the effect of SNPs across k traits using z-scores from previous regression analyses. It performs simulations to produce p-values, using the empirical multivariate-normal distribution of null z-scores.                                                                     | Julia      |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/kaskarn/JaSPU)                                           |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| MTAR            | 2020 | [Paper](https://doi.org/10.1038/s41467-020-16591-0)                                  | Joint analysis of association summary statistics between multiple rare variants and different traits. Leverages the genome-wide genetic correlation to inform the degree of gene-level effect heterogeneity across traits.                                                             | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [CRAN](https://cran.r-project.org/web/packages/MTAR/)                                |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| JASS            | 2020 | [Paper](https://doi.org/10.1093/nargab/lqaa003)                                      | Incorporates various joint tests such as the omnibus approach and weighted sum of Z-score tests while offering data cleaning and harmonization, fast derivation of joint statistics, and optimized data management process.                                                            | Python/web |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitLab](https://gitlab.pasteur.fr/statistical-genetics/jass)                        |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| PLACO           | 2020 | [Paper](https://doi.org/10.1371/journal.pgen.1009218)                                | Implements a variant-level formal statistical test of pleiotropy of two traits inspired from mediation analysis.                                                                                                                                                                       | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/RayDebashree/PLACO)                                      |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| HOPS            | 2019 | [Paper](https://doi.org/10.1186/s13059-019-1844-7)                                   | Allows to compute the horizontal pleiotropy score by removing correlations between traits caused by vertical pleiotropy and normalizing effect sizes across all traits.                                                                                                                | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/rondolab/HOPS)                                           |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| MSKAT           | 2019 | [Paper](https://doi.org/10.1093/bioinformatics/bty811)                               | Various types of multi-trait SNP-set association tests (variance component test, burden test and adaptive test), and efficient numerical calculation of p-values.                                                                                                                      | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/baolinwu/MSKAT)                                          |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| multiSKAT       | 2019 | [Paper](https://doi.org/10.1002/gepi.22156)                                          | A general framework for testing pleiotropic effects of rare variants on multiple continuous phenotypes using multivariate kernel regression. Many existing tests are equivalent to specific choices of parameters within this framework.                                               | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/diptavo/MultiSKAT)                                       |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| bmass           | 2019 | [Paper](https://doi.org/10.1371/journal.pgen.1008431)                                | Bayesian multivariate analysis of GWAS data using univariate association summary statistics.                                                                                                                                                                                           | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/mturchin20/bmass)                                        |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| MTAR            | 2019 | [Paper](https://doi.org/10.1093/bioinformatics/bty961)                               | Uses principal component (PC)-based association test which has optimal power when the underlying multi-trait signal can be captured by the first PC. Performs an adaptive test by optimally weighting the PC-based test and the omnibus chi-square test to achieve robust performance. | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/baolinwu/MTAR)                                           |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| HIPO            | 2018 | [Paper](https://doi.org/10.1371/journal.pgen.1007549)                                | Performs heritability informed power optimization for conducting multi-trait association analysis.                                                                                                                                                                                     | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/gqi/hipo)                                                |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| MTAG            | 2018 | [Paper](https://doi.org/10.1038/s41588-017-0009-4)                                   | A method for joint analysis of GWAS of different traits, using a weigthed sum and LDSC.                                                                                                                                                                                                | Python     |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/JonJala/mtag)                                            |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| UNITY           | 2018 | [Paper](https://doi.org/10.1093/bioinformatics/bty254)                               | A Bayesian framework for estimating the proportion of causal variants shared between a pair of complex traits.                                                                                                                                                                         | Python     |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/bogdanlab/UNITY)                                         |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| CONFIT          | 2018 | [Paper](https://doi.org/10.1093/bioinformatics/bty249)                               | The method estimates the degree of shared effects between traits from the data. The test statistic is a sum of the relative likelihoods for each alternate configuration.                                                                                                              | Python     |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/lgai/CONFIT)                                             |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| iMAP            | 2018 | [Paper](https://doi.org/10.1093/bioinformatics/bty204)                               | Performs integrative mapping of pleiotropic association and functional annotations using penalized Gaussian mixture models. Uses a multinomial logistic regression model.                                                                                                              | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/biostatpzeng/iMAP)                                       |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| Plei            | 2017 | [Paper](https://doi.org/10.1534/genetics.117.300347)                                 | A procedure that can be applied for both marginal analysis and conditional analysis. Uses the union-intersection testing methods, but in addition to the likelihood ratio test, it also applies generalized estimating equations under the working independence model.                 | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/yangq001/Plei)                                           |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| EPS             | 2016 | [Paper](https://doi.org/10.1093/bioinformatics/btw081)                               | An Empirical Bayes approach to integrating Pleiotropy and Tissue-Specific information (EPS) for prioritizing risk genes.                                                                                                                                                               | Matlab     |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/gordonliu810822/EPS)                                     |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| USAT            | 2016 | [Paper](https://doi.org/10.1002/gepi.21937)                                          | Uses a data-adaptive weighted score-based test statistic for testing association of multiple continuous phenotypes with a single SNP.                                                                                                                                                  | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/RayDebashree/USAT)                                       |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| gwas-pw         | 2016 | [Paper](https://doi.org/10.1038/ng.3570)                                             | A tool for jointly analysing two GWAS to identify loci that influence both traits. Instead of using two P-value thresholds to identify variants that influence both traits, the algorithm learns reasonable thresholds from the data.                                                  | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/joepickrell/gwas-pw)                                     |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| cFDR            | 2015 | [Paper](https://doi.org/10.1371/journal.pgen.1004926)                                | Calculates an upper bound on the expected false discovery rate (FDR) across a set of SNPs whose p-values for two diseases are both less than two disease-specific threshold.                                                                                                           | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/jamesliley/cFDR-common-controls)                         |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| MGAS            | 2015 | [Paper](https://doi.org/10.1093/bioinformatics/btu783)                               | A multivariate gene-based association test by extended Simes procedure (MGAS), that allows gene-based testing of multivariate phenotypes.                                                                                                                                              | Java       |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [Webpage](https://pmglab.top/kgg/)                                                   |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| GPA             | 2014 | [Paper](https://doi.org/10.1371/journal.pgen.1004787)                                | Uses the EM algorithm to integrate pleiotropy and functional annotation (eQTL etc).                                                                                                                                                                                                    | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [GitHub](https://github.com/dongjunchung/GPA)                                        |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| TATES           | 2013 | [Paper](https://doi.org/10.1371/journal.pgen.1003235)                                | Trait-based Association Test that uses Extended Simes procedure combines the p-values obtained in standard univariate GWAS to acquire one trait-based p-value, while correcting for correlations between components.                                                                   | Fortran    |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [Webpage](https://cncr.nl/research/tates/)                                           |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+
| p_ACT           | 2007 | [Paper](https://doi.org/10.1086/522036)                                              | A method of computing p-values adjusted for correlated tests that attains the accuracy of permutation or simulation-based tests in much less computation time.                                                                                                                         | R          |
|                 |      |                                                                                      |                                                                                                                                                                                                                                                                                        |            |
|                 |      | [Webpage](https://csg.sph.umich.edu/boehnke/p_act.php)                               |                                                                                                                                                                                                                                                                                        |            |
+-----------------+------+--------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+

: Software are sorted by most to least recent {.striped}

### Mendelian randomization

+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| Name           | Year     | Links                                                                             | Description                                                                                                                                                                                                                                                                    | Language |
+================+==========+===================================================================================+================================================================================================================================================================================================================================================================================+==========+
| TS_LMM         | Preprint | [Paper](https://doi.org/10.1101/2023.04.25.23289099)                              | Performs two-stage linear mixed model for MVMR that accounts for variance of summary statistics not only in outcome, but also in all of the risk factors.                                                                                                                      | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/mingding-hsph/LMM_MEC)                                |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MRBEE          | 2024     | [Paper](https://doi.org/10.1016/j.xhgg.2024.100290)                               | A multivariable MR method capable of simultaneously removing measurement error bias and identifying horizontal pleiotropy.                                                                                                                                                     | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/noahlorinczcomi/MRBEE)                                |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MVMR-cML       | 2023     | [Paper](https://doi.org/10.1016/j.ajhg.2023.02.014)                               | An efficient and robust MVMR method based on constrained maximum likelihood (cML).                                                                                                                                                                                             | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/ZhaotongL/MVMR-cML)                                   |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MRlap          | 2023     | [Paper](https://doi.org/10.1002/gepi.22522)                                       | Simultaneously considers weak instrument bias and winner's curse while accounting for potential sample overlap and corrects IVW-MR.                                                                                                                                            | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/n-mounier/MRlap)                                      |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MRCI           | 2023     | [Paper](https://doi.org/10.1038/s41467-023-36490-4)                               | Estimates reciprocal causation between two phenotypes simultaneously using reference LD information.                                                                                                                                                                           | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/zpliu/MRCI)                                           |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MR²            | 2023     | [Paper](https://doi.org/10.1016/j.ajhg.2023.06.005)                               | Performs MR for multiple outcomes to identify exposures that cause more than one outcome or, conversely, exposures that exert their effect on distinct responses.                                                                                                              | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/lb664/MR2/)                                           |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| pIVW           | 2023     | [Paper](https://doi.org/10.1111/biom.13732)                                       | An extension to IVW that accounts for weak instruments and balanced horizontal pleiotropy simultaneously.                                                                                                                                                                      | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [CRAN](https://cran.r-project.org/web/packages/mr.pivw/index.html)                |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| BiDirectCausal | 2022     | [Paper](https://doi.org/10.1371/journal.pgen.1010205)                             | Infers possible bi-directional causal effects between two traits.                                                                                                                                                                                                              | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/xue-hr/BiDirectCausal)                                |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MR-Corr2       | 2022     | [Paper](https://doi.org/10.1093/bioinformatics/btab646)                           | A Bayesian approach that uses the orthogonal projection to reparameterize the bivariate normal distribution for effects of variants on exposure and horizontal pleiotropy.                                                                                                     | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/QingCheng0218/MR.Corr2)                               |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MR.CUE         | 2022     | [Paper](https://doi.org/10.1038/s41467-022-34164-1)                               | Estimates causal effect while identifying IVs with correlated horizontal pleiotropy and accounting for estimation uncertainty.                                                                                                                                                 | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/QingCheng0218/MR.CUE)                                 |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| adOMICs        | 2022     | [Paper](https://doi.org/10.1093/bib/bbac376)                                      | Used to investigate the causal effects of multiple omics biomarkers on an outcome. The method first tests the effect of each omics biomarker on the outcome separately using an MR method and then combines the p-values using various methods.                                | Python   |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/lshen/adomics)                                        |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MRLocus        | 2021     | [Paper](https://doi.org/10.1371/journal.pgen.1009455)                             | A method for Bayesian estimation of the gene-to-trait effect from eQTL and GWAS data for loci with evidence of allelic heterogeneity. Makes use of a colocalization step applied to each nearly-LD-independent eQTL, followed by an MR analysis step across eQTLs.             | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://thelovelab.github.io/mrlocus/)                                   |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| OMR            | 2021     | [Paper](https://doi.org/10.1093/bib/bbab322)                                      | MR method that uses all GWAS SNPs for causal inference. The method accommodates the commonly encountered horizontal pleiotropy effects and relies on a composite likelihood framework for scalable computation.                                                                | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/wanglu205/OMR)                                        |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| JAM-MR         | 2021     | [Paper](https://doi.org/10.1002/sim.9109)                                         | Performs variable selection and causal effect estimation in MR as an extension of the JAM algorithm.                                                                                                                                                                           | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/pjnewcombe/R2BGLiMS)                                  |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| hJAM           | 2021     | [Paper](https://doi.org/10.1093/aje/kwaa287)                                      | A two-stage hierarchical model that unifies the framework of MR and TWAS and can be applied to correlated instruments and multiple intermediates.                                                                                                                              | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/USCbiostats/hJAM)                                     |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| IMRP           | 2021     | [Paper](https://doi.org/10.1093/bioinformatics/btaa985)                           | Performs Iterative MR and Pleiotropy analysis to simultaneously search for horizontal pleiotropic variants and estimate causal effect.                                                                                                                                         | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/XiaofengZhuCase/IMRP)                                 |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MRcML          | 2021     | [Paper](https://doi.org/10.1016/j.ajhg.2021.05.014)                               | Uses constrained maximum likelihood and model averaging that is robust to invalid IVs with uncorrelated or correlated pleiotropic effects.                                                                                                                                     | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/xue-hr/MRcML)                                         |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| LHC-MR         | 2021     | [Paper](https://doi.org/10.1038/s41467-021-26970-w)                               | Estimates bi-directional causal effects, direct heritabilities, and confounder effects while accounting for sample overlap.                                                                                                                                                    | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/LizaDarrous/lhcMR)                                    |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| Mr.MtRobin     | 2021     | [Paper](https://doi.org/10.1002/gepi.22380)                                       | Multi-tissue transcriptome-wide MR method that uses multi-tissue eQTL analyses as input and a reverse regression random slope mixed model to infer whether a gene is associated with a complex trait.                                                                          | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/kjgleason/Mr.MtRobin)                                 |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MR-LDP         | 2020     | [Paper](https://doi.org/10.1093/nargab/lqaa028)                                   | A probabilistic model for MR in the presence of LD and to account for horizontal pleiotropy.                                                                                                                                                                                   | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/QingCheng0218/MR.LDP)                                 |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| BWMR           | 2020     | [Paper](https://doi.org/10.1093/bioinformatics/btz749)                            | Bayesian methods for MR.                                                                                                                                                                                                                                                       | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/jiazhao97/BWMR)                                       |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MRCD           | 2020     | [Paper](https://doi.org/10.1371/journal.pgen.1009105)                             | Infers causal direction between two traits in the presence of horizontal pleiotropy.                                                                                                                                                                                           | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/xue-hr/MRCD)                                          |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MR-BMA         | 2020     | [Paper](https://doi.org/10.1038/s41467-019-13870-3)                               | Bayesian algorithm to perform risk factor selection in multivariable MR.                                                                                                                                                                                                       | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/verena-zuber/demo_AMD)                                |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| PMR            | 2020     | [Paper](https://doi.org/10.1038/s41467-020-17668-6)                               | Efficient inference of 2SMR in TWAS. It can account for correlated instruments and horizontal pleiotropy, and provides accurate estimates of causal effect and horizontal pleiotropy effect. It can be applied in single traits as well as multiple correlated outcome traits. | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/yuanzhongshang/PMR)                                   |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MR             | 2020     | [Paper](https://doi.org/10.1038/s41467-019-14156-4)                               | Several standard methods (simple and weighted median, IVW, and MR-Egger) for performing MR analyses with summary data.                                                                                                                                                         | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [CRAN](https://cran.r-project.org/web/packages/MendelianRandomization/index.html) |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MRMix          | 2019     | [Paper](https://doi.org/10.1038/s41467-019-09432-2)                               | MR analysis using an underlying mixture model incorporating a fraction of the genetic instruments to have direct effect on the outcome (horizontal pleiotropy).                                                                                                                | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/gqi/MRMix)                                            |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| TWMR           | 2019     | [Paper](https://doi.org/10.1038/s41467-019-10936-0)                               | MR integrating GWAS and eQTL data reveals genetic determinants of complex and clinical traits.                                                                                                                                                                                 | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/eleporcu/TWMR)                                        |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MVMR           | 2019     | [Paper](https://doi.org/10.1093/ije/dyy262)                                       | Performs multivariable MR analyses, including heterogeneity statistics for assessing instrument strength and validity.                                                                                                                                                         | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://wspiller.github.io/MVMR/)                                        |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MR.RAPS        | 2019     | [Paper](https://doi.org/10.1093/ije/dyz142)                                       | A three-sample genome-wide design with many independent genetic instruments across the whole genome. The method is efficient with many weak genetic instruments and robust to balanced and/or sparse pleiotropy.                                                               | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/qingyuanzhao/mr.raps)                                 |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MR-PRESSO      | 2018     | [Paper](https://doi.org/10.1038/s41588-018-0099-7)                                | Allows for the evaluation of horizontal pleiotropy in multi-instrument MR.                                                                                                                                                                                                     | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/rondolab/MR-PRESSO)                                   |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MR-Base        | 2018     | [Paper](https://doi.org/10.7554/elife.34408)                                      | A database and web-based analytical platform for Mendelian randomization. It is coupled to TwoSampleMR and to MRC IEU OpenGWAS database.                                                                                                                                       | R/web    |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [Webpage](https://www.mrbase.org/)                                                |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| LCV            | 2018     | [Paper](https://doi.org/10.1038/s41588-018-0255-0)                                | Estimates causal associations between traits avoiding confounding by genetic correlation, using LDSC.                                                                                                                                                                          | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://github.com/lukejoconnor/LCV)                                     |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MRPEA          | 2018     | [Paper](https://doi.org/10.1093/bib/bbx025)                                       | A pathway association MR analysis approach, which was capable of correcting the genetic confounding effects of environmental exposures, using data of environmental exposures.                                                                                                 | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [Forge](https://sourceforge.net/projects/mrpea/)                                  |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| TwoSampleMR    | 2017     | [Paper](https://doi.org/10.1371/journal.pgen.1007081)                             | Standard methods for performing MR. It uses the IEU GWAS database and to the MRBase web app.                                                                                                                                                                                   | R        |
|                |          |                                                                                   |                                                                                                                                                                                                                                                                                |          |
|                |          | [GitHub](https://mrcieu.github.io/TwoSampleMR/)                                   |                                                                                                                                                                                                                                                                                |          |
+----------------+----------+-----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+

: Software are sorted by most to least recent {.striped}

### Colocalization

+--------------+------+---------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| Name         | Year | Links                                                                     | Description                                                                                                                                                                                                                                                                                                                                       | Language |
+==============+======+===========================================================================+===================================================================================================================================================================================================================================================================================================================================================+==========+
| SharePro     | 2024 | [Paper](https://doi.org/10.1093/bioinformatics/btae295)                   | Takes an effect group-level approach to integrate LD modelling and colocalization assessment to account for multiple causal variants in colocalization analysis.                                                                                                                                                                                  | Python   |
|              |      |                                                                           |                                                                                                                                                                                                                                                                                                                                                   |          |
|              |      | [GitHub](https://github.com/zhwm/SharePro_coloc)                          |                                                                                                                                                                                                                                                                                                                                                   |          |
+--------------+------+---------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| SS2          | 2022 | [Paper](https://doi.org/10.1016/j.ajhg.2021.12.012)                       | Integrates GWAS summary statistics with eQTL summary statistics across any number of gene-by-tissue pairs, is applicable when there are overlapping participants in the two studies.                                                                                                                                                              | R        |
|              |      |                                                                           |                                                                                                                                                                                                                                                                                                                                                   |          |
|              |      | [GitHub](https://github.com/FanWang0216/SimpleSum2Colocalization)         |                                                                                                                                                                                                                                                                                                                                                   |          |
+--------------+------+---------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| ColocQuiaL   | 2022 | [Paper](https://doi.org/10.1093/bioinformatics/btac512)                   | A pipeline that provides a framework to perform eQTL and sQTL colocalization analyses at scale across the genome with COLOC.                                                                                                                                                                                                                      | R        |
|              |      |                                                                           |                                                                                                                                                                                                                                                                                                                                                   |          |
|              |      | [GitHub](https://github.com/bvoightlab/ColocQuiaL)                        |                                                                                                                                                                                                                                                                                                                                                   |          |
+--------------+------+---------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MSG          | 2022 | [Paper](https://doi.org/10.1371/journal.pgen.1009814)                     | A multidimensional splicing gene approach.                                                                                                                                                                                                                                                                                                        | R        |
|              |      |                                                                           |                                                                                                                                                                                                                                                                                                                                                   |          |
|              |      | [GitHub](https://github.com/yingji15/MSG_public)                          |                                                                                                                                                                                                                                                                                                                                                   |          |
+--------------+------+---------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| ezQTL        | 2022 | [Paper](https://doi.org/10.1016/j.gpb.2022.05.004)                        | Performs data quality control for variants matched between different datasets, LD visualization, and two-trait colocalization analyses using two state-of-the-art methodologies (eCAVIAR and HyPrColoc).                                                                                                                                          | R/web    |
|              |      |                                                                           |                                                                                                                                                                                                                                                                                                                                                   |          |
|              |      | [Webpage](https://analysistools.cancer.gov/ezqtl/#/home)                  |                                                                                                                                                                                                                                                                                                                                                   |          |
+--------------+------+---------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| COLOC        | 2021 | [Paper](https://doi.org/10.1371/journal.pgen.1009440)                     | Allows evidence for association at multiple causal variants to be evaluated simultaneously, whilst separating the statistical support for each variant conditional on the causal signal being considered.                                                                                                                                         | R        |
|              |      |                                                                           |                                                                                                                                                                                                                                                                                                                                                   |          |
|              |      | [GitHub](https://github.com/chr1swallace/coloc)                           |                                                                                                                                                                                                                                                                                                                                                   |          |
+--------------+------+---------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| POEMColoc    | 2021 | [Paper](https://doi.org/10.1186/s12859-021-04170-z)                       | An approximation to the coloc method that can be applied when limited summary statistics are available.                                                                                                                                                                                                                                           | R        |
|              |      |                                                                           |                                                                                                                                                                                                                                                                                                                                                   |          |
|              |      | [GitHub](https://github.com/AbbVie-ComputationalGenomics/POEMColoc)       |                                                                                                                                                                                                                                                                                                                                                   |          |
+--------------+------+---------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| HyPrColoc    | 2021 | [Paper](https://doi.org/10.1038/s41467-020-20885-8)                       | An efficient deterministic Bayesian algorithm that can detect colocalization across vast numbers of traits simultaneously.                                                                                                                                                                                                                        | R        |
|              |      |                                                                           |                                                                                                                                                                                                                                                                                                                                                   |          |
|              |      | [GitHub](https://github.com/cnfoley/hyprcoloc)                            |                                                                                                                                                                                                                                                                                                                                                   |          |
+--------------+------+---------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| LocusFocus   | 2020 | [Paper](https://doi.org/10.1371/journal.pcbi.1008336)                     | A web-based tool that tests colocalization using the Simple Sum method to identify the most relevant genes and tissues for a particular locus in the presence of high LD and/or allelic heterogeneity.                                                                                                                                            | web      |
|              |      |                                                                           |                                                                                                                                                                                                                                                                                                                                                   |          |
|              |      | [Webpage](https://locusfocus.research.sickkids.ca/)                       |                                                                                                                                                                                                                                                                                                                                                   |          |
+--------------+------+---------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| SparkINFERNO | 2020 | [Paper](https://doi.org/10.1093/bioinformatics/btaa246)                   | A scalable bioinformatics pipeline characterizing non-coding GWAS. It prioritizes causal variants underlying association signals and reports relevant regulatory elements, tissue contexts and plausible target genes.                                                                                                                            | Python   |
|              |      |                                                                           |                                                                                                                                                                                                                                                                                                                                                   |          |
|              |      | [Bitbucket](https://bitbucket.org/wanglab-upenn/sparkinferno/src/master/) |                                                                                                                                                                                                                                                                                                                                                   |          |
+--------------+------+---------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| pwCoCo       | 2020 | [Paper](https://doi.org/10.1038/s41588-020-0682-6)                        | A fast tool that integrates methods from GCTA-COJO and the coloc R package.                                                                                                                                                                                                                                                                       | Python   |
|              |      |                                                                           |                                                                                                                                                                                                                                                                                                                                                   |          |
|              |      | [GitHub](https://github.com/jwr-git/pwcoco)                               |                                                                                                                                                                                                                                                                                                                                                   |          |
+--------------+------+---------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| PESCA        | 2020 | [Paper](https://doi.org/10.1016/j.ajhg.2020.04.012)                       | Uses ancestry-matched estimates of LD to infer genome-wide proportions of population-specific and shared causal variants for a single trait in two populations. These estimates are then used as priors in an empirical Bayes framework to localize and test for enrichment of population-specific/shared causal variants in regions of interest. | C/C++    |
|              |      |                                                                           |                                                                                                                                                                                                                                                                                                                                                   |          |
|              |      | [GitHub](https://huwenboshi.github.io/pesca/)                             |                                                                                                                                                                                                                                                                                                                                                   |          |
+--------------+------+---------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| moloc        | 2018 | [Paper](https://doi.org/10.1093/bioinformatics/bty147)                    | Multiple-trait-coloc, a Bayesian statistical framework that integrates GWAS summary data with multiple molecular QTL data to identify regulatory effects at GWAS risk loci.                                                                                                                                                                       | R        |
|              |      |                                                                           |                                                                                                                                                                                                                                                                                                                                                   |          |
|              |      | [GitHub](https://github.com/clagiamba/moloc)                              |                                                                                                                                                                                                                                                                                                                                                   |          |
+--------------+------+---------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| LLR          | 2017 | [Paper](https://doi.org/10.1093/bioinformatics/btx512)                    | A latent low-rank approach to colocalizing genetic risk variants in multiple GWAS and phenotypes.                                                                                                                                                                                                                                                 | Matlab   |
|              |      |                                                                           |                                                                                                                                                                                                                                                                                                                                                   |          |
|              |      | [GitHub](https://github.com/gordonliu810822/LLR)                          |                                                                                                                                                                                                                                                                                                                                                   |          |
+--------------+------+---------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| eCAVIAR      | 2016 | [Paper](https://doi.org/10.1016/j.ajhg.2016.10.003)                       | Colocalization of GWAS and eQTL with a probabilistic method that accounts for more than one causal variant in any given locus.                                                                                                                                                                                                                    | C/C++    |
|              |      |                                                                           |                                                                                                                                                                                                                                                                                                                                                   |          |
|              |      | [GitHub](https://github.com/fhormoz/caviar)                               |                                                                                                                                                                                                                                                                                                                                                   |          |
+--------------+------+---------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| Sherlock     | 2013 | [Paper](https://doi.org/10.1016/j.ajhg.2013.03.022)                       | Uses a database of eQTL n different tissues to identify patterns in GWAS that match those for specific genes. information from both cis- and trans- eQTL SNPs.                                                                                                                                                                                    | web      |
|              |      |                                                                           |                                                                                                                                                                                                                                                                                                                                                   |          |
|              |      | [Webpage](http://sherlock.ucsf.edu/)                                      |                                                                                                                                                                                                                                                                                                                                                   |          |
+--------------+------+---------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+

: Software are sorted by most to least recent {.striped}

### TWAS

+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| Name        | Year | Links                                                             | Description                                                                                                                                                                                                                                                                    | Language |
+=============+======+===================================================================+================================================================================================================================================================================================================================================================================+==========+
| OPERA       | 2023 | [Paper](https://doi.org/10.1016/j.xgen.2023.100344)               | A method that jointly analyzes GWAS and multi-omics xQTL data to enhance the identification of molecular phenotypes associated with complex traits through shared causal variants.                                                                                             | C/C++    |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/wuyangf7/OPERA)                       |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| TScML       | 2023 | [Paper](https://doi.org/10.1080/01621459.2023.2183127)            | A robust and efficient inferential method to account for both hidden confounding and some invalid IVs via two-stage constrained maximum likelihood, an extension of 2SLS.                                                                                                      | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/xue-hr/TScML)                         |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| SUMMIT      | 2022 | [Paper](https://doi.org/10.1038/s41467-022-34016-y)               | Improves the expression prediction model accuracy and the power of TWAS by using a large eQTL summary-level dataset, penalized regression and Cauchy Combination Test.                                                                                                         | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/ChongWuLab/SUMMIT)                    |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| ARCHIE      | 2022 | [Paper](https://doi.org/10.1038/s41467-022-31845-9)               | A summary statistic based sparse canonical correlation analysis method to identify sets of gene-expressions trans-regulated by sets of known trait-related genetic variants.                                                                                                   | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/diptavo/ARCHIE)                       |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| MA-FOCUS    | 2022 | [Paper](https://doi.org/10.1016/j.ajhg.2022.07.002)               | An extension of FOCUS that leverages summary GWAS data with eQTL weights from multiple ancestries to increase the precision of credible sets for causal genes.                                                                                                                 | Python   |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/mancusolab/ma-focus)                  |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| CoMM-S⁴     | 2021 | [Paper](https://doi.org/10.3389/fgene.2021.704538)                | Uses variational Bayesian EM algorithm and a likelihood ratio test to integrate GWAS data with eQTL to assess expression-trait association.                                                                                                                                    | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/gordonliu810822/CoMM)                 |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| sCCA        | 2021 | [Paper](https://doi.org/10.1371/journal.pgen.1008973)             | Integrates multiple tissues in the TWAS using sparse canonical correlation analysis and an aggregate Cauchy association test.                                                                                                                                                  | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/fenghelian/sCCA-ACAT_TWAS)            |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| Mr.MtRobin  | 2021 | [Paper](https://doi.org/10.1002/gepi.22380)                       | Multi-tissue transcriptome-wide MR method that uses multi-tissue eQTL analyses as input and a reverse regression random slope mixed model to infer whether a gene is associated with a complex trait.                                                                          | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/kjgleason/Mr.MtRobin)                 |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| HMAT        | 2021 | [Paper](https://doi.org/10.1093/hmg/ddab056)                      | A method which aggregates TWAS association evidence obtained across multiple gene expression prediction models by leveraging the harmonic mean p-value combination.                                                                                                            | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/biostatpzeng/HMAT)                    |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| BGW-TWAS    | 2020 | [Paper](https://doi.org/10.1016/j.ajhg.2020.08.022)               | A Bayesian TWAS method that leverages both cis- and trans-eQTL based on Bayesian variable selection regression.                                                                                                                                                                | C/C++    |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/yanglab-emory/BGW-TWAS)               |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| BAGEA       | 2020 | [Paper](https://doi.org/10.1371/journal.pcbi.1007770)             | A variational Bayes framework to model cis-eQTLs using directed and undirected genomic annotations.                                                                                                                                                                            | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/dlampart/bagea)                       |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| TisCoMM     | 2020 | [Paper](https://doi.org/10.1093/nar/gkaa767)                      | Leverages the co-regulation of genetic variations across different tissues explicitly via a probabilistic model. Apart from prioritizing gene-trait associations, it also detects the tissue-specific role of candidate target genes in complex traits.                        | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/XingjieShi/TisCoMM)                   |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| Primo       | 2020 | [Paper](https://doi.org/10.1186/s13059-020-02125-w)               | A method for integrative analysis of multiple sets of xQTL data (eQTL, pQTL etc) from different cellular conditions or studies. It examines association patterns of SNPs to complex and omics traits accounting for LD, heterogeneity and sample correlations.                 | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/kjgleason/Primo)                      |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| PMR         | 2020 | [Paper](https://doi.org/10.1038/s41467-020-17668-6)               | Efficient inference of 2SMR in TWAS. It can account for correlated instruments and horizontal pleiotropy, and provides accurate estimates of causal effect and horizontal pleiotropy effect. It can be applied in single traits as well as multiple correlated outcome traits. | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/yuanzhongshang/PMR)                   |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| JEPEGMIX2‐P | 2020 | [Paper](https://doi.org/10.1002/ajmg.b.32823)                     | A fast TWAS pathway method that has uses a large and diverse reference panel and is applicable to ethnically mixed-cohorts.                                                                                                                                                    | Exe      |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/Chatzinakos/JEPEGMIX2-P)              |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| PTWAS       | 2020 | [Paper](https://doi.org/10.1186/s13059-020-02026-y)               | A probabilistic TWAS method which applies principles from instrumental variables analysis and takes advantage of probabilistic eQTL annotations.                                                                                                                               | Exe      |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/xqwen/ptwas)                          |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| sMiST       | 2020 | [Paper](https://doi.org/10.1371/journal.pgen.1008947)             | Mixed effects score test that tests for the total effect of both the effect of the mediator by imputing genetically predicted gene expression.                                                                                                                                 | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [Webpage](https://research.fredhutch.org/hsu/en/software.html)    |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| iFunMed     | 2019 | [Paper](https://doi.org/10.1002/gepi.22217)                       | A mediation model that utilizes functional annotation and statistics from GWAS and eQTL. It enables identification of SNPs that are associated with phenotypical changes through direct phenotype-genotype and/or indirect phenotype-genotype through gene expression effect.  | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/mcrojo/iFunMed)                       |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| UTMOST      | 2019 | [Paper](https://doi.org/10.1038/s41588-019-0345-7)                | A framework that combines multiple single-tissue associations into a powerful metric to quantify the overall gene-trait association.                                                                                                                                           | Python   |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/Joker-Jerome/UTMOST)                  |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| TIGAR       | 2019 | [Paper](https://doi.org/10.1016/j.ajhg.2019.05.018)               | Transcriptome-Wide Association Studies (TWAS) by training gene expression imputation models by nonparametric Bayesian Dirichlet Process Regression (DPR) and Elastic-Net (PrediXcan) methods with reference transcriptomic panels.                                             | Python   |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/yanglab-emory/TIGAR)                  |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| TWMR        | 2019 | [Paper](https://doi.org/10.1038/s41467-019-10936-0)               | MR integrating GWAS and eQTL data reveals genetic determinants of complex and clinical traits.                                                                                                                                                                                 | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/eleporcu/TWMR)                        |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| FOCUS       | 2019 | [Paper](https://doi.org/10.1038/s41588-019-0367-1)                | Software to fine-map transcriptome-wide association study statistics at genomic risk regions. The software outputs a credible set of genes to explain observed genomic risk.                                                                                                   | Python   |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/mancusolab/ma-focus)                  |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| GSMR        | 2018 | [Paper](https://doi.org/10.1038/s41467-017-02317-2)               | Generalised Summary-data-based Mendelian Randomisation method that tests for a putative causal association between two phenotypes based on a multi-SNP model.                                                                                                                  | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [Webpage](https://yanglab.westlake.edu.cn/software/gsmr/)         |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| S-MultiXcan | 2018 | [Paper](https://doi.org/10.1038/s41467-018-03621-1)               | It integrates summary statistics from multiple single-tissue transcriptome-wide association studies (TWAS) to identify genes whose expression is associated with complex traits.                                                                                               | Python   |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/hakyimlab/MetaXcan)                   |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| S-PrediXcan | 2018 | [Paper](https://doi.org/10.1038/s41467-018-03621-1)               | A method that seeks to capture the effects of gene expression variation on human phenotypes.                                                                                                                                                                                   | Python   |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/hakyimlab/MetaXcan)                   |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| TWAS-aSPU   | 2017 | [Paper](https://doi.org/10.1534/genetics.117.300270)              | Implements the so-called sum of powered score (SPU) which includes sum and SSU as special cases.                                                                                                                                                                               | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [Webpage](https://wuchong.org/TWAS.html)                          |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| fQTL        | 2017 | [Paper](https://doi.org/10.1101/107623)                           | A multi-tissue, multivariate model for mapping expression quantitative trait loci and predicting gene expression. Technically in pre-print but since 2017.                                                                                                                     | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [GitHub](https://github.com/ypark/fqtl)                           |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| SMR         | 2016 | [Paper](https://doi.org/10.1038/ng.3538)                          | Integrates GWAS with eQTL to identify genes whose expression levels are associated with a complex trait because of pleiotropy.                                                                                                                                                 | Exe      |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [Webpage](https://yanglab.westlake.edu.cn/software/smr/#Overview) |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| FUSION      | 2016 | [Paper](https://doi.org/10.1038/ng.3506)                          | integrates gene expression measurements with summary association statistics from large-scale genome-wide association studies (GWAS) to identify genes whose cis-regulated expression is associated to complex traits.                                                          | R        |
|             |      |                                                                   |                                                                                                                                                                                                                                                                                |          |
|             |      | [Webpage](http://gusevlab.org/projects/fusion/)                   |                                                                                                                                                                                                                                                                                |          |
+-------------+------+-------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+

: Software are sorted by most to least recent {.striped}

## References

<ul class="fa-ul">

<li class="li-pad">

[<i class="fa-solid fa-file-pen"></i>]{.fa-li} Kontou, P. I., & Bagos, P. G. (2024). The goldmine of GWAS summary statistics: A systematic review of methods and tools. *BioData Mining*, *17*(31). <https://doi.org/10.1186/s13040-024-00385-x><br> [Open Access {{<ai open-access>}}]{.badge .bg-primary} [[Download PDF {{<fa file-pdf>}}](https://biodatamining.biomedcentral.com/counter/pdf/10.1186/s13040-024-00385-x.pdf)]{.badge .bg-secondary .bg-link}

</li>

</ul>
